{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-04-16T08:07:00.687345Z",
     "start_time": "2025-04-16T08:07:00.384613Z"
    }
   },
   "source": "from utils import *",
   "outputs": [],
   "execution_count": 1
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T08:21:47.784315Z",
     "start_time": "2025-04-16T08:21:47.742571Z"
    }
   },
   "cell_type": "code",
   "source": [
    "pdxnet_sheet = '/Users/tushar/Downloads/PDXNetAllModels (4).tsv'\n",
    "pdxnet_df = read_metadata_with_fields(pdxnet_sheet)\n",
    "wustl_df = pdxnet_df[pdxnet_df['Contributor'] == 'WUSTL'].reset_index(drop=True)\n",
    "wustl_df['model_id'] = wustl_df['ContributorPDX.ID'].str.split('; ')\n",
    "wustl_df = wustl_df.explode('model_id').reset_index(drop=True)\n",
    "wustl_df['model_id'] = wustl_df['model_id'].str.split(', ')\n",
    "wustl_df = wustl_df.explode('model_id').reset_index(drop=True)\n",
    "wustl_df['model_id'] = wustl_df['model_id'].str.split(';')\n",
    "wustl_df = wustl_df.explode('model_id').reset_index(drop=True)\n",
    "wustl_df['model_id'] = wustl_df['model_id'].str.split(',')\n",
    "wustl_df = wustl_df.explode('model_id').reset_index(drop=True)\n",
    "wustl_df['model_id'] = [ f[0] for f in wustl_df['model_id'].str.split(' ')]"
   ],
   "id": "ad8b8d15262113b5",
   "outputs": [],
   "execution_count": 39
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T08:21:49.554165Z",
     "start_time": "2025-04-16T08:21:49.526407Z"
    }
   },
   "cell_type": "code",
   "source": "wustl_df[wustl_df['model_id'].duplicated(keep=False)]",
   "id": "28df8b280d6a8528",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                     PDXSource Contributor    ContributorPDX.ID  \\\n",
       "9    PDXNet Consortium Members       WUSTL  WU-0010-T1 (001-R1)   \n",
       "184  PDXNet Consortium Members       WUSTL           WU-0010-T1   \n",
       "\n",
       "    PDMR.Patient.ID Gender  CTEP.SDCCode      CTEP.SDCDescription  \\\n",
       "9            K20605   Male      10038045  Adenocarcinoma - rectum   \n",
       "184          K96138   Male      10038045  Adenocarcinoma - rectum   \n",
       "\n",
       "    Diagnosis.Subtype       Disease.Body.Location Age.at.Diagnosis  ...  \\\n",
       "9                   -  Digestive/Gastrointestinal               60  ...   \n",
       "184          mucinous  Digestive/Gastrointestinal               60  ...   \n",
       "\n",
       "    Grade.Stage.Information  \\\n",
       "9                       TNM   \n",
       "184      TNM (Pathological)   \n",
       "\n",
       "                                         Patient.Notes Molecular.and.IHC.Data  \\\n",
       "9    Tumor Grade/Stage: T3N1bM1b, moderately differ...                      -   \n",
       "184  Tumor Grade/Stage: Moderately differentiated, ...                      -   \n",
       "\n",
       "    Has.Smoked.100.Cigarettes          Race               Ethnicity  \\\n",
       "9                Not Provided         White  Not Hispanic or Latino   \n",
       "184                       Yes  Not Provided  Not Hispanic or Latino   \n",
       "\n",
       "                           Additional.Medical.History OncoTree.Code  \\\n",
       "9                                                   -           NaN   \n",
       "184  Family History: 1st Degree Relative, Skin cancer  NOT_SELECTED   \n",
       "\n",
       "    Submission    model_id  \n",
       "9    2/26/2019  WU-0010-T1  \n",
       "184   1/5/2023  WU-0010-T1  \n",
       "\n",
       "[2 rows x 22 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PDXSource</th>\n",
       "      <th>Contributor</th>\n",
       "      <th>ContributorPDX.ID</th>\n",
       "      <th>PDMR.Patient.ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>CTEP.SDCCode</th>\n",
       "      <th>CTEP.SDCDescription</th>\n",
       "      <th>Diagnosis.Subtype</th>\n",
       "      <th>Disease.Body.Location</th>\n",
       "      <th>Age.at.Diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>Grade.Stage.Information</th>\n",
       "      <th>Patient.Notes</th>\n",
       "      <th>Molecular.and.IHC.Data</th>\n",
       "      <th>Has.Smoked.100.Cigarettes</th>\n",
       "      <th>Race</th>\n",
       "      <th>Ethnicity</th>\n",
       "      <th>Additional.Medical.History</th>\n",
       "      <th>OncoTree.Code</th>\n",
       "      <th>Submission</th>\n",
       "      <th>model_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PDXNet Consortium Members</td>\n",
       "      <td>WUSTL</td>\n",
       "      <td>WU-0010-T1 (001-R1)</td>\n",
       "      <td>K20605</td>\n",
       "      <td>Male</td>\n",
       "      <td>10038045</td>\n",
       "      <td>Adenocarcinoma - rectum</td>\n",
       "      <td>-</td>\n",
       "      <td>Digestive/Gastrointestinal</td>\n",
       "      <td>60</td>\n",
       "      <td>...</td>\n",
       "      <td>TNM</td>\n",
       "      <td>Tumor Grade/Stage: T3N1bM1b, moderately differ...</td>\n",
       "      <td>-</td>\n",
       "      <td>Not Provided</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/26/2019</td>\n",
       "      <td>WU-0010-T1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>PDXNet Consortium Members</td>\n",
       "      <td>WUSTL</td>\n",
       "      <td>WU-0010-T1</td>\n",
       "      <td>K96138</td>\n",
       "      <td>Male</td>\n",
       "      <td>10038045</td>\n",
       "      <td>Adenocarcinoma - rectum</td>\n",
       "      <td>mucinous</td>\n",
       "      <td>Digestive/Gastrointestinal</td>\n",
       "      <td>60</td>\n",
       "      <td>...</td>\n",
       "      <td>TNM (Pathological)</td>\n",
       "      <td>Tumor Grade/Stage: Moderately differentiated, ...</td>\n",
       "      <td>-</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Not Provided</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>Family History: 1st Degree Relative, Skin cancer</td>\n",
       "      <td>NOT_SELECTED</td>\n",
       "      <td>1/5/2023</td>\n",
       "      <td>WU-0010-T1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 22 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 40
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T08:22:08.370447Z",
     "start_time": "2025-04-16T08:22:08.361827Z"
    }
   },
   "cell_type": "code",
   "source": "wustl_df.columns",
   "id": "3af3929c1c28f5ac",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['PDXSource', 'Contributor', 'ContributorPDX.ID', 'PDMR.Patient.ID',\n",
       "       'Gender', 'CTEP.SDCCode', 'CTEP.SDCDescription', 'Diagnosis.Subtype',\n",
       "       'Disease.Body.Location', 'Age.at.Diagnosis', 'Date.of.Diagnosis',\n",
       "       'Has.Known.Metastatic.Disease', 'Grade.Stage.Information',\n",
       "       'Patient.Notes', 'Molecular.and.IHC.Data', 'Has.Smoked.100.Cigarettes',\n",
       "       'Race', 'Ethnicity', 'Additional.Medical.History', 'OncoTree.Code',\n",
       "       'Submission', 'model_id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 41
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T08:27:47.813129Z",
     "start_time": "2025-04-16T08:27:47.789680Z"
    }
   },
   "cell_type": "code",
   "source": "wustl_df.to_csv('/Users/tushar/Downloads/PDXNetAllModels_models.tsv', sep='\\t', index=False)",
   "id": "b7a294155c142681",
   "outputs": [],
   "execution_count": 48
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:28:09.113389Z",
     "start_time": "2025-04-16T09:28:08.525856Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup\n",
    "from requests import get"
   ],
   "id": "ca5c631ab5256395",
   "outputs": [],
   "execution_count": 52
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:34:51.746847Z",
     "start_time": "2025-04-16T09:34:51.026660Z"
    }
   },
   "cell_type": "code",
   "source": [
    "url = 'https://pdx.wustl.edu/pdx/'\n",
    "response = get(url)\n",
    "soup = BeautifulSoup(response.text, 'html.parser')\n",
    "ctypes_table = soup.find_all('table' , {'id': 'ctypes'})[0]\n",
    "data = []\n",
    "for row in ctypes_table.find_all('tr')[1:-1]:  # skip header and total row\n",
    "    cells = row.find_all('td')\n",
    "    cancer_type = cells[0].get_text(strip=True)\n",
    "    link = cells[0].find('a')['href']\n",
    "    description = cells[1].get_text(strip=True)\n",
    "    number_of_models = int(cells[2].get_text(strip=True))\n",
    "    data.append({\n",
    "        'type': cancer_type,\n",
    "        'link': link,\n",
    "        'description': description,\n",
    "        'models': number_of_models\n",
    "    })"
   ],
   "id": "cf4e7c6000f8497e",
   "outputs": [],
   "execution_count": 79
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T10:15:10.174231Z",
     "start_time": "2025-04-16T10:15:10.170589Z"
    }
   },
   "cell_type": "code",
   "source": "from tqdm import tqdm",
   "id": "b0b1f7ae19909fba",
   "outputs": [],
   "execution_count": 137
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T10:15:21.215794Z",
     "start_time": "2025-04-16T10:15:16.195566Z"
    }
   },
   "cell_type": "code",
   "source": [
    "final_df = pd.DataFrame()\n",
    "for d in tqdm(data):\n",
    "    new_url = url + d['link']\n",
    "    response = get(new_url)\n",
    "    soup = BeautifulSoup(response.text, 'html.parser')\n",
    "    table = soup.find('table', {'class': 'center'})\n",
    "\n",
    "    # Extract headers\n",
    "    headers = [th.get_text(strip=True).replace('\\n', ' ') for th in table.find_all('th')]\n",
    "\n",
    "    # Extract rows\n",
    "    rows = []\n",
    "    for tr in table.find_all('tr')[1:]:\n",
    "        tds = tr.find_all('td')\n",
    "        row = []\n",
    "        for i, td in enumerate(tds):\n",
    "            if i == 8:  # PDX Samples\n",
    "                # Join all span text with commas or line breaks\n",
    "                samples = [span.get_text(strip=True) for span in td.find_all('span')]\n",
    "                row.append(\", \".join(samples))\n",
    "            elif i == 9:  # Genetic Alterations (might be empty or have links/text)\n",
    "                links = [a['href'] for a in td.find_all('a', href=True)]\n",
    "                row.append(\", \".join(links))\n",
    "            else:\n",
    "                row.append(td.get_text(strip=True))\n",
    "        if row:\n",
    "            rows.append(row)\n",
    "\n",
    "    # Create DataFrame\n",
    "    df = pd.DataFrame(rows, columns=headers)\n",
    "    df['type'] = d['type']\n",
    "    final_df = pd.concat([final_df, df])\n",
    "final_df"
   ],
   "id": "6c39025d29def5af",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 15%|█▌        | 4/26 [00:04<00:25,  1.14s/it]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mKeyboardInterrupt\u001B[0m                         Traceback (most recent call last)",
      "Cell \u001B[0;32mIn[138], line 4\u001B[0m\n\u001B[1;32m      2\u001B[0m \u001B[38;5;28;01mfor\u001B[39;00m d \u001B[38;5;129;01min\u001B[39;00m tqdm(data):\n\u001B[1;32m      3\u001B[0m     new_url \u001B[38;5;241m=\u001B[39m url \u001B[38;5;241m+\u001B[39m d[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mlink\u001B[39m\u001B[38;5;124m'\u001B[39m]\n\u001B[0;32m----> 4\u001B[0m     response \u001B[38;5;241m=\u001B[39m \u001B[43mget\u001B[49m\u001B[43m(\u001B[49m\u001B[43mnew_url\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m      5\u001B[0m     soup \u001B[38;5;241m=\u001B[39m BeautifulSoup(response\u001B[38;5;241m.\u001B[39mtext, \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mhtml.parser\u001B[39m\u001B[38;5;124m'\u001B[39m)\n\u001B[1;32m      6\u001B[0m     table \u001B[38;5;241m=\u001B[39m soup\u001B[38;5;241m.\u001B[39mfind(\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mtable\u001B[39m\u001B[38;5;124m'\u001B[39m, {\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mclass\u001B[39m\u001B[38;5;124m'\u001B[39m: \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mcenter\u001B[39m\u001B[38;5;124m'\u001B[39m})\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/api.py:73\u001B[0m, in \u001B[0;36mget\u001B[0;34m(url, params, **kwargs)\u001B[0m\n\u001B[1;32m     62\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21mget\u001B[39m(url, params\u001B[38;5;241m=\u001B[39m\u001B[38;5;28;01mNone\u001B[39;00m, \u001B[38;5;241m*\u001B[39m\u001B[38;5;241m*\u001B[39mkwargs):\n\u001B[1;32m     63\u001B[0m \u001B[38;5;250m    \u001B[39m\u001B[38;5;124mr\u001B[39m\u001B[38;5;124;03m\"\"\"Sends a GET request.\u001B[39;00m\n\u001B[1;32m     64\u001B[0m \n\u001B[1;32m     65\u001B[0m \u001B[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001B[39;00m\n\u001B[0;32m   (...)\u001B[0m\n\u001B[1;32m     70\u001B[0m \u001B[38;5;124;03m    :rtype: requests.Response\u001B[39;00m\n\u001B[1;32m     71\u001B[0m \u001B[38;5;124;03m    \"\"\"\u001B[39;00m\n\u001B[0;32m---> 73\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43mrequest\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mget\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mparams\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mparams\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/api.py:59\u001B[0m, in \u001B[0;36mrequest\u001B[0;34m(method, url, **kwargs)\u001B[0m\n\u001B[1;32m     55\u001B[0m \u001B[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001B[39;00m\n\u001B[1;32m     56\u001B[0m \u001B[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001B[39;00m\n\u001B[1;32m     57\u001B[0m \u001B[38;5;66;03m# cases, and look like a memory leak in others.\u001B[39;00m\n\u001B[1;32m     58\u001B[0m \u001B[38;5;28;01mwith\u001B[39;00m sessions\u001B[38;5;241m.\u001B[39mSession() \u001B[38;5;28;01mas\u001B[39;00m session:\n\u001B[0;32m---> 59\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43msession\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[43m(\u001B[49m\u001B[43mmethod\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mmethod\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43murl\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:587\u001B[0m, in \u001B[0;36mSession.request\u001B[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001B[0m\n\u001B[1;32m    582\u001B[0m send_kwargs \u001B[38;5;241m=\u001B[39m {\n\u001B[1;32m    583\u001B[0m     \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mtimeout\u001B[39m\u001B[38;5;124m\"\u001B[39m: timeout,\n\u001B[1;32m    584\u001B[0m     \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mallow_redirects\u001B[39m\u001B[38;5;124m\"\u001B[39m: allow_redirects,\n\u001B[1;32m    585\u001B[0m }\n\u001B[1;32m    586\u001B[0m send_kwargs\u001B[38;5;241m.\u001B[39mupdate(settings)\n\u001B[0;32m--> 587\u001B[0m resp \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msend\u001B[49m\u001B[43m(\u001B[49m\u001B[43mprep\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43msend_kwargs\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    589\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m resp\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:701\u001B[0m, in \u001B[0;36mSession.send\u001B[0;34m(self, request, **kwargs)\u001B[0m\n\u001B[1;32m    698\u001B[0m start \u001B[38;5;241m=\u001B[39m preferred_clock()\n\u001B[1;32m    700\u001B[0m \u001B[38;5;66;03m# Send the request\u001B[39;00m\n\u001B[0;32m--> 701\u001B[0m r \u001B[38;5;241m=\u001B[39m \u001B[43madapter\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msend\u001B[49m\u001B[43m(\u001B[49m\u001B[43mrequest\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    703\u001B[0m \u001B[38;5;66;03m# Total elapsed time of the request (approximately)\u001B[39;00m\n\u001B[1;32m    704\u001B[0m elapsed \u001B[38;5;241m=\u001B[39m preferred_clock() \u001B[38;5;241m-\u001B[39m start\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/adapters.py:489\u001B[0m, in \u001B[0;36mHTTPAdapter.send\u001B[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001B[0m\n\u001B[1;32m    487\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m    488\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m chunked:\n\u001B[0;32m--> 489\u001B[0m         resp \u001B[38;5;241m=\u001B[39m \u001B[43mconn\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43murlopen\u001B[49m\u001B[43m(\u001B[49m\n\u001B[1;32m    490\u001B[0m \u001B[43m            \u001B[49m\u001B[43mmethod\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmethod\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    491\u001B[0m \u001B[43m            \u001B[49m\u001B[43murl\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    492\u001B[0m \u001B[43m            \u001B[49m\u001B[43mbody\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mbody\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    493\u001B[0m \u001B[43m            \u001B[49m\u001B[43mheaders\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mheaders\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    494\u001B[0m \u001B[43m            \u001B[49m\u001B[43mredirect\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[1;32m    495\u001B[0m \u001B[43m            \u001B[49m\u001B[43massert_same_host\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[1;32m    496\u001B[0m \u001B[43m            \u001B[49m\u001B[43mpreload_content\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[1;32m    497\u001B[0m \u001B[43m            \u001B[49m\u001B[43mdecode_content\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[1;32m    498\u001B[0m \u001B[43m            \u001B[49m\u001B[43mretries\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmax_retries\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    499\u001B[0m \u001B[43m            \u001B[49m\u001B[43mtimeout\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mtimeout\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    500\u001B[0m \u001B[43m        \u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    502\u001B[0m     \u001B[38;5;66;03m# Send the request.\u001B[39;00m\n\u001B[1;32m    503\u001B[0m     \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m    504\u001B[0m         \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mhasattr\u001B[39m(conn, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mproxy_pool\u001B[39m\u001B[38;5;124m\"\u001B[39m):\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:715\u001B[0m, in \u001B[0;36mHTTPConnectionPool.urlopen\u001B[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001B[0m\n\u001B[1;32m    712\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_prepare_proxy(conn)\n\u001B[1;32m    714\u001B[0m \u001B[38;5;66;03m# Make the request on the httplib connection object.\u001B[39;00m\n\u001B[0;32m--> 715\u001B[0m httplib_response \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_make_request\u001B[49m\u001B[43m(\u001B[49m\n\u001B[1;32m    716\u001B[0m \u001B[43m    \u001B[49m\u001B[43mconn\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    717\u001B[0m \u001B[43m    \u001B[49m\u001B[43mmethod\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    718\u001B[0m \u001B[43m    \u001B[49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    719\u001B[0m \u001B[43m    \u001B[49m\u001B[43mtimeout\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mtimeout_obj\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    720\u001B[0m \u001B[43m    \u001B[49m\u001B[43mbody\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mbody\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    721\u001B[0m \u001B[43m    \u001B[49m\u001B[43mheaders\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mheaders\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    722\u001B[0m \u001B[43m    \u001B[49m\u001B[43mchunked\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mchunked\u001B[49m\u001B[43m,\u001B[49m\n\u001B[1;32m    723\u001B[0m \u001B[43m\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    725\u001B[0m \u001B[38;5;66;03m# If we're going to release the connection in ``finally:``, then\u001B[39;00m\n\u001B[1;32m    726\u001B[0m \u001B[38;5;66;03m# the response doesn't need to know about the connection. Otherwise\u001B[39;00m\n\u001B[1;32m    727\u001B[0m \u001B[38;5;66;03m# it will also try to release it and we'll have a double-release\u001B[39;00m\n\u001B[1;32m    728\u001B[0m \u001B[38;5;66;03m# mess.\u001B[39;00m\n\u001B[1;32m    729\u001B[0m response_conn \u001B[38;5;241m=\u001B[39m conn \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m release_conn \u001B[38;5;28;01melse\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:467\u001B[0m, in \u001B[0;36mHTTPConnectionPool._make_request\u001B[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001B[0m\n\u001B[1;32m    462\u001B[0m             httplib_response \u001B[38;5;241m=\u001B[39m conn\u001B[38;5;241m.\u001B[39mgetresponse()\n\u001B[1;32m    463\u001B[0m         \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mBaseException\u001B[39;00m \u001B[38;5;28;01mas\u001B[39;00m e:\n\u001B[1;32m    464\u001B[0m             \u001B[38;5;66;03m# Remove the TypeError from the exception chain in\u001B[39;00m\n\u001B[1;32m    465\u001B[0m             \u001B[38;5;66;03m# Python 3 (including for exceptions like SystemExit).\u001B[39;00m\n\u001B[1;32m    466\u001B[0m             \u001B[38;5;66;03m# Otherwise it looks like a bug in the code.\u001B[39;00m\n\u001B[0;32m--> 467\u001B[0m             \u001B[43msix\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mraise_from\u001B[49m\u001B[43m(\u001B[49m\u001B[43me\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;28;43;01mNone\u001B[39;49;00m\u001B[43m)\u001B[49m\n\u001B[1;32m    468\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m (SocketTimeout, BaseSSLError, SocketError) \u001B[38;5;28;01mas\u001B[39;00m e:\n\u001B[1;32m    469\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_raise_timeout(err\u001B[38;5;241m=\u001B[39me, url\u001B[38;5;241m=\u001B[39murl, timeout_value\u001B[38;5;241m=\u001B[39mread_timeout)\n",
      "File \u001B[0;32m<string>:3\u001B[0m, in \u001B[0;36mraise_from\u001B[0;34m(value, from_value)\u001B[0m\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:462\u001B[0m, in \u001B[0;36mHTTPConnectionPool._make_request\u001B[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001B[0m\n\u001B[1;32m    459\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mTypeError\u001B[39;00m:\n\u001B[1;32m    460\u001B[0m     \u001B[38;5;66;03m# Python 3\u001B[39;00m\n\u001B[1;32m    461\u001B[0m     \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m--> 462\u001B[0m         httplib_response \u001B[38;5;241m=\u001B[39m \u001B[43mconn\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mgetresponse\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    463\u001B[0m     \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mBaseException\u001B[39;00m \u001B[38;5;28;01mas\u001B[39;00m e:\n\u001B[1;32m    464\u001B[0m         \u001B[38;5;66;03m# Remove the TypeError from the exception chain in\u001B[39;00m\n\u001B[1;32m    465\u001B[0m         \u001B[38;5;66;03m# Python 3 (including for exceptions like SystemExit).\u001B[39;00m\n\u001B[1;32m    466\u001B[0m         \u001B[38;5;66;03m# Otherwise it looks like a bug in the code.\u001B[39;00m\n\u001B[1;32m    467\u001B[0m         six\u001B[38;5;241m.\u001B[39mraise_from(e, \u001B[38;5;28;01mNone\u001B[39;00m)\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:1349\u001B[0m, in \u001B[0;36mHTTPConnection.getresponse\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m   1347\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m   1348\u001B[0m     \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m-> 1349\u001B[0m         \u001B[43mresponse\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mbegin\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m   1350\u001B[0m     \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mConnectionError\u001B[39;00m:\n\u001B[1;32m   1351\u001B[0m         \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mclose()\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:316\u001B[0m, in \u001B[0;36mHTTPResponse.begin\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m    314\u001B[0m \u001B[38;5;66;03m# read until we get a non-100 response\u001B[39;00m\n\u001B[1;32m    315\u001B[0m \u001B[38;5;28;01mwhile\u001B[39;00m \u001B[38;5;28;01mTrue\u001B[39;00m:\n\u001B[0;32m--> 316\u001B[0m     version, status, reason \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_read_status\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    317\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m status \u001B[38;5;241m!=\u001B[39m CONTINUE:\n\u001B[1;32m    318\u001B[0m         \u001B[38;5;28;01mbreak\u001B[39;00m\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:277\u001B[0m, in \u001B[0;36mHTTPResponse._read_status\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m    276\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21m_read_status\u001B[39m(\u001B[38;5;28mself\u001B[39m):\n\u001B[0;32m--> 277\u001B[0m     line \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mstr\u001B[39m(\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mfp\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mreadline\u001B[49m\u001B[43m(\u001B[49m\u001B[43m_MAXLINE\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m+\u001B[39;49m\u001B[43m \u001B[49m\u001B[38;5;241;43m1\u001B[39;49m\u001B[43m)\u001B[49m, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124miso-8859-1\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[1;32m    278\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mlen\u001B[39m(line) \u001B[38;5;241m>\u001B[39m _MAXLINE:\n\u001B[1;32m    279\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m LineTooLong(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mstatus line\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/socket.py:704\u001B[0m, in \u001B[0;36mSocketIO.readinto\u001B[0;34m(self, b)\u001B[0m\n\u001B[1;32m    702\u001B[0m \u001B[38;5;28;01mwhile\u001B[39;00m \u001B[38;5;28;01mTrue\u001B[39;00m:\n\u001B[1;32m    703\u001B[0m     \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m--> 704\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_sock\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mrecv_into\u001B[49m\u001B[43m(\u001B[49m\u001B[43mb\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    705\u001B[0m     \u001B[38;5;28;01mexcept\u001B[39;00m timeout:\n\u001B[1;32m    706\u001B[0m         \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_timeout_occurred \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;01mTrue\u001B[39;00m\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1241\u001B[0m, in \u001B[0;36mSSLSocket.recv_into\u001B[0;34m(self, buffer, nbytes, flags)\u001B[0m\n\u001B[1;32m   1237\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m flags \u001B[38;5;241m!=\u001B[39m \u001B[38;5;241m0\u001B[39m:\n\u001B[1;32m   1238\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m(\n\u001B[1;32m   1239\u001B[0m           \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mnon-zero flags not allowed in calls to recv_into() on \u001B[39m\u001B[38;5;132;01m%s\u001B[39;00m\u001B[38;5;124m\"\u001B[39m \u001B[38;5;241m%\u001B[39m\n\u001B[1;32m   1240\u001B[0m           \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m\u001B[38;5;18m__class__\u001B[39m)\n\u001B[0;32m-> 1241\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[43mnbytes\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mbuffer\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m   1242\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m   1243\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28msuper\u001B[39m()\u001B[38;5;241m.\u001B[39mrecv_into(buffer, nbytes, flags)\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1099\u001B[0m, in \u001B[0;36mSSLSocket.read\u001B[0;34m(self, len, buffer)\u001B[0m\n\u001B[1;32m   1097\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m   1098\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m buffer \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[0;32m-> 1099\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_sslobj\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;28;43mlen\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mbuffer\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m   1100\u001B[0m     \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m   1101\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_sslobj\u001B[38;5;241m.\u001B[39mread(\u001B[38;5;28mlen\u001B[39m)\n",
      "\u001B[0;31mKeyboardInterrupt\u001B[0m: "
     ]
    }
   ],
   "execution_count": 138
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:46:39.608266Z",
     "start_time": "2025-04-16T09:46:39.596086Z"
    }
   },
   "cell_type": "code",
   "source": [
    "wustl_df_mol_data = final_df[final_df['Source'] == 'WUSTL'].reset_index(drop=True)\n",
    "wustl_df_mol_data = wustl_df_mol_data[['ModelName', 'Source', 'Patient', 'Patient Race(Self-reported)', 'PatientGender', 'Specimen TreatmentPrior to Collection', 'PDX Samples', 'GeneticAlterations']]\n",
    "wustl_df_mol_data.to_csv('wustl_mol_data.tsv', sep='\\t', index=False)"
   ],
   "id": "1090b1a53a58bb3f",
   "outputs": [],
   "execution_count": 92
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-17T08:15:13.909399Z",
     "start_time": "2025-04-17T08:15:13.696403Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mutation_template = read_metadata_with_fields(join(templates, 'mut/mutation_template_external-mutation_template_external.tsv'))\n",
    "cna_template = read_metadata_with_fields(join(templates, 'cna/cna_template-sheet.tsv'))\n",
    "mms = read_metadata_with_fields(join(home, 'WUSTL/WUSTL_molecular_metadata-sample.tsv'))"
   ],
   "id": "c0b67944b129e358",
   "outputs": [],
   "execution_count": 140
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T10:09:30.232566Z",
     "start_time": "2025-04-16T10:09:30.223639Z"
    }
   },
   "cell_type": "code",
   "source": "mms.columns",
   "id": "9c4e5caba69aff04",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Field', 'model_id', 'sample_id', 'sample_origin', 'passage',\n",
       "       'host_strain_name', 'host_strain_nomenclature',\n",
       "       'engrafted_tumor_collection_site', 'raw_data_url', 'platform_id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 131
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-17T08:36:45.189622Z",
     "start_time": "2025-04-17T08:15:25.037782Z"
    }
   },
   "cell_type": "code",
   "source": [
    "final_mut_df = pd.DataFrame()\n",
    "\n",
    "for _, row in tqdm(wustl_df_mol_data.iterrows()):\n",
    "    model_id = row['ModelName']\n",
    "    sample_ids = row['PDX Samples'].split(', ')\n",
    "    links = row['GeneticAlterations'].split(', ')\n",
    "    sid_link_dict = dict(zip(sample_ids, links))\n",
    "    for sid in sid_link_dict:\n",
    "        new_url = url + sid_link_dict[sid]\n",
    "        response = get(new_url)\n",
    "        soup = BeautifulSoup(response.text, 'html.parser')\n",
    "        table = soup.find_all('table')[1]\n",
    "        # Extract headers\n",
    "        headers = [th.get_text(strip=True).replace('\\n', ' ') for th in table.find_all('th')]\n",
    "\n",
    "        # Extract rows\n",
    "        rows = []\n",
    "        for tr in soup.find_all('tr')[1:]:\n",
    "            cells = tr.find_all('td')\n",
    "            row = [td.get_text(strip=True).replace('\\n', '') for td in cells]\n",
    "            if row:\n",
    "                rows.append(row)\n",
    "\n",
    "        # Create DataFrame\n",
    "\n",
    "        df = pd.DataFrame(rows, columns=headers).iloc[3:]\n",
    "\n",
    "        temp_cna = df[df['CN status']!= '-']\n",
    "        cna_df = cna_template\n",
    "        cna_df['symbol'] = temp_cna['Gene']\n",
    "        cna_df['copy_number_status'] = temp_cna['CN status']\n",
    "        cna_df['gistic_value'] = temp_cna['SomaticCopy Number']\n",
    "        cna_df['sample_id'] = sid\n",
    "        cna_df['platform_id'] = 'wustl_cna_status'\n",
    "\n",
    "        temp_mut = df[df['SomaticMutations']!= '-']\n",
    "        if temp_mut.shape[0] > 0:\n",
    "            mut_df = mutation_template\n",
    "            mut_df['symbol'] = temp_mut['Gene']\n",
    "            mut_df['amino_acid_change'] = temp_mut['SomaticMutations']\n",
    "            mut_df['allele_frequency'] = temp_mut['VAF']\n",
    "            mut_df['consequence'] = temp_mut['Classification']\n",
    "            mut_df['variant_class'] = temp_mut['Type']\n",
    "            mut_df['sample_id'] = sid\n",
    "            mut_df['platform_id'] = 'wustl_mutation'\n",
    "            final_mut_df = pd.concat([final_mut_df, mut_df])\n",
    "\n",
    "            new_row_df = pd.DataFrame([['', model_id, sid, 'xenograft', '0', 'NSG', 'NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</sup>/SzJ', '', '', 'wustl_mutation']], columns=mms.columns)\n",
    "            mms = pd.concat([mms, new_row_df], ignore_index=True)\n",
    "\n",
    "        #cna_df.to_csv(f'cna/WUSTL_cna_{sid}.tsv', sep='\\t', index=False)\n",
    "        #new_row_df = pd.DataFrame([['', model_id, sid, 'xenograft', '0', 'NSG', 'NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</sup>/SzJ', '', '', 'wustl_cna_status']], columns=mms.columns)\n",
    "        #mms = pd.concat([mms, new_row_df], ignore_index=True)\n",
    "mms.to_csv(join(home, 'WUSTL/WUSTL_molecular_metadata-sample_2.tsv'), sep='\\t', index=False)"
   ],
   "id": "816defd662921df7",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "133it [21:20,  9.62s/it]\n"
     ]
    }
   ],
   "execution_count": 141
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T09:34:52.744522Z",
     "start_time": "2025-04-22T09:34:52.440688Z"
    }
   },
   "cell_type": "code",
   "source": "final_mut_df.to_csv('WUSLT_mut.tsv', sep='\\t', index=False)",
   "id": "53b4ef3f9951b07",
   "outputs": [],
   "execution_count": 171
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:29:58.785871Z",
     "start_time": "2025-04-22T10:29:58.744581Z"
    }
   },
   "cell_type": "code",
   "source": "final_mut_df.head()",
   "id": "4000efce7e378271",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           sample_id    symbol read_depth allele_frequency chromosome strand  \\\n",
       "281  WU-0029-T1-X1P0   ADAMTS7        NaN            0.513        NaN    NaN   \n",
       "284  WU-0029-T1-X1P0  ADAMTSL1        NaN            0.472        NaN    NaN   \n",
       "329  WU-0029-T1-X1P0    ADRA2C        NaN            0.733        NaN    NaN   \n",
       "693  WU-0029-T1-X1P0    ARID1A        NaN            0.682        NaN    NaN   \n",
       "870  WU-0029-T1-X1P0      BAG5        NaN            0.077        NaN    NaN   \n",
       "\n",
       "    seq_start_position seq_end_position ref_allele alt_allele variation_id  \\\n",
       "281                NaN              NaN        NaN        NaN          NaN   \n",
       "284                NaN              NaN        NaN        NaN          NaN   \n",
       "329                NaN              NaN        NaN        NaN          NaN   \n",
       "693                NaN              NaN        NaN        NaN          NaN   \n",
       "870                NaN              NaN        NaN        NaN          NaN   \n",
       "\n",
       "    platform amino_acid_change consequence variant_class     platform_id  \n",
       "281      NaN          p.R1635C    Missense           SNP  wustl_mutation  \n",
       "284      NaN           p.E238*    Nonsense           SNP  wustl_mutation  \n",
       "329      NaN           p.E410K    Missense           SNP  wustl_mutation  \n",
       "693      NaN           p.Q557*    Nonsense           SNP  wustl_mutation  \n",
       "870      NaN           p.E419K    Missense           SNP  wustl_mutation  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>read_depth</th>\n",
       "      <th>allele_frequency</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>seq_start_position</th>\n",
       "      <th>seq_end_position</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>platform</th>\n",
       "      <th>amino_acid_change</th>\n",
       "      <th>consequence</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>platform_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>WU-0029-T1-X1P0</td>\n",
       "      <td>ADAMTS7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.513</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.R1635C</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>wustl_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>WU-0029-T1-X1P0</td>\n",
       "      <td>ADAMTSL1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.E238*</td>\n",
       "      <td>Nonsense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>wustl_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>WU-0029-T1-X1P0</td>\n",
       "      <td>ADRA2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.733</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.E410K</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>wustl_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>WU-0029-T1-X1P0</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.Q557*</td>\n",
       "      <td>Nonsense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>wustl_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>WU-0029-T1-X1P0</td>\n",
       "      <td>BAG5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.E419K</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>wustl_mutation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 190
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:31:13.821766Z",
     "start_time": "2025-04-22T10:31:13.812594Z"
    }
   },
   "cell_type": "code",
   "source": "aa_change_df = final_mut_df.drop_duplicates(['symbol', 'amino_acid_change'])[['symbol', 'amino_acid_change']]",
   "id": "5da7bf5c113e3d9c",
   "outputs": [],
   "execution_count": 196
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:31:18.311952Z",
     "start_time": "2025-04-22T10:31:18.300731Z"
    }
   },
   "cell_type": "code",
   "source": "aa_change_df",
   "id": "75e879b177213f7",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        symbol    amino_acid_change\n",
       "281    ADAMTS7             p.R1635C\n",
       "284   ADAMTSL1              p.E238*\n",
       "329     ADRA2C              p.E410K\n",
       "693     ARID1A              p.Q557*\n",
       "870       BAG5              p.E419K\n",
       "...        ...                  ...\n",
       "3108      TPP1         p.H166Mfs*17\n",
       "2619   SMPDL3A               p.T51S\n",
       "329        APC  p.K1310Dfs*4p.R213*\n",
       "3108     GPR37              p.R482W\n",
       "693      ASXL3             p.P1943S\n",
       "\n",
       "[99 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>symbol</th>\n",
       "      <th>amino_acid_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>ADAMTS7</td>\n",
       "      <td>p.R1635C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>ADAMTSL1</td>\n",
       "      <td>p.E238*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>ADRA2C</td>\n",
       "      <td>p.E410K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>p.Q557*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>BAG5</td>\n",
       "      <td>p.E419K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3108</th>\n",
       "      <td>TPP1</td>\n",
       "      <td>p.H166Mfs*17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2619</th>\n",
       "      <td>SMPDL3A</td>\n",
       "      <td>p.T51S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>APC</td>\n",
       "      <td>p.K1310Dfs*4p.R213*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3108</th>\n",
       "      <td>GPR37</td>\n",
       "      <td>p.R482W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>ASXL3</td>\n",
       "      <td>p.P1943S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 197
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T12:11:15.169118Z",
     "start_time": "2025-04-22T12:08:03.951713Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import requests\n",
    "import csv\n",
    "\n",
    "# Input: list of (Gene Symbol, Amino Acid Change)\n",
    "variants = [\n",
    "    (\"BRAF\", \"p.V600E\"),\n",
    "    (\"TP53\", \"p.R175H\"),\n",
    "    (\"CDH1\", \"p.R598Q\")\n",
    "    # add more here\n",
    "]\n",
    "\n",
    "vcf_lines = [\n",
    "    \"##fileformat=VCFv4.2\",\n",
    "    \"#CHROM\\tPOS\\tSTOP\\tID\\tREF\\tALT\\tcdna\\taa_change\\tgene\\tQUAL\\tFILTER\\tINFO\"\n",
    "]\n",
    "\n",
    "def get_hgnc_symbol(symbol):\n",
    "    url = \"https://mygene.info/v3/query\"\n",
    "    params = {\n",
    "        \"q\": symbol,\n",
    "        \"species\": \"human\",\n",
    "        \"fields\": \"symbol,name,HGNC\",\n",
    "        \"size\": 2\n",
    "    }\n",
    "    response = requests.get(url, params=params)\n",
    "    data = response.json()\n",
    "    if data[\"hits\"]:\n",
    "        if len(data[\"hits\"]) > 1:\n",
    "            return [data[\"hits\"][0][\"symbol\"], data[\"hits\"][1][\"symbol\"]]\n",
    "        else:\n",
    "            return [data[\"hits\"][0][\"symbol\"], '']\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "def get_clinvar_variants(gene, aa_change):\n",
    "    search_term = f\"{gene}[gene]+{aa_change}[varname]\"\n",
    "    url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"clinvar\",\n",
    "        \"term\": search_term,\n",
    "        \"retmode\": \"json\",\n",
    "        \"retmax\": \"1\",\n",
    "        \"retmode\": \"json\",\n",
    "        \"rettype\": \"clinvarset\"\n",
    "    }\n",
    "\n",
    "    resp = requests.get(url, params=params)\n",
    "    resp.raise_for_status()\n",
    "    data = resp.json()\n",
    "\n",
    "    ids = data.get(\"esearchresult\", {}).get(\"idlist\", [])\n",
    "    return ids\n",
    "\n",
    "def fetch_clinvar_record(variant_id):\n",
    "    summary_url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"clinvar\",\n",
    "        \"id\": variant_id,\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "\n",
    "    resp = requests.get(summary_url, params=params)\n",
    "    resp.raise_for_status()\n",
    "    record = resp.json()[\"result\"][variant_id]\n",
    "    return record\n",
    "\n",
    "def parse_cdna_change(cdna_change):\n",
    "    import re\n",
    "    match = re.match(r\"c\\.(\\d+)([ACGT])>([ACGT])\", cdna_change)\n",
    "    if match:\n",
    "        ref = match.group(2)\n",
    "        alt = match.group(3)\n",
    "        return ref, alt\n",
    "    else:\n",
    "        raise ValueError(f\"Invalid cDNA change format: {cdna_change}\")\n",
    "\n",
    "\n",
    "\n",
    "def extract_vcf_info(record, gene, aa_change):\n",
    "    if \"variation_set\" in record:\n",
    "        var = record[\"variation_set\"][0]\n",
    "        cdna_change = var['cdna_change']\n",
    "        for placement in var.get(\"variation_loc\", []):\n",
    "            if placement.get(\"assembly_name\") == \"GRCh38\":\n",
    "                    chr = placement['chr']\n",
    "                    start_pos = placement[\"start\"]\n",
    "                    end_pos = placement[\"stop\"]\n",
    "                    ref, alt = parse_cdna_change(cdna_change)\n",
    "                    return chr, start_pos, end_pos,\".\", ref, alt, cdna_change, gene, aa_change\n",
    "    return None\n",
    "\n",
    "# Loop through input variants\n",
    "for _, row in aa_change_df.iterrows():\n",
    "    gene, aa_change = row\n",
    "    print(f\"Looking up {gene} {aa_change}...\")\n",
    "    ids = get_clinvar_variants(gene, aa_change)\n",
    "    if not ids:\n",
    "        hits = get_hgnc_symbol(gene)\n",
    "        if hits is not None:\n",
    "            ids = get_clinvar_variants(hits[0], aa_change)\n",
    "            if (not ids or len(ids) == 0) and hits[1]!= '':\n",
    "                ids = get_clinvar_variants(hits[1], aa_change)\n",
    "                if not ids or len(ids) == 0:\n",
    "                    print(f\"❌ No ClinVar match for {gene} {aa_change}\")\n",
    "                    continue\n",
    "    if len(ids) == 0:\n",
    "        continue\n",
    "    record = fetch_clinvar_record(ids[0])\n",
    "    #print(record)\n",
    "    vcf_info = extract_vcf_info(record, gene, aa_change)\n",
    "    print(vcf_info)\n",
    "    if vcf_info:\n",
    "        vcf_lines.append(\"\\t\".join(map(str, list(vcf_info) + [\".\", \".\", \".\"])))\n",
    "    else:\n",
    "        print(f\"⚠️ Could not extract VCF info for {gene} {aa_change}\")\n",
    "\n",
    "# Write to VCF file\n",
    "with open(\"clinvar_variants.vcf\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(vcf_lines))\n",
    "\n",
    "print(\"✅ VCF written to clinvar_variants.vcf\")\n"
   ],
   "id": "3fb8044778602a08",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "❌ No ClinVar match for ADAMTS7 p.R1635C\n",
      "❌ No ClinVar match for ARID1A p.Q557*\n",
      "❌ No ClinVar match for BRINP3 p.K600T\n",
      "❌ No ClinVar match for CDC42EP1 p.A256T\n",
      "('16', '68822082', '68822082', '.', 'G', 'A', 'c.1793G>A', 'CDH1', 'p.R598Q')\n",
      "❌ No ClinVar match for CENPB p.S188L\n",
      "❌ No ClinVar match for DSCAM p.E1093K\n",
      "❌ No ClinVar match for ERBB3 p.S1234F\n",
      "('20', '6077214', '6077214', '.', 'G', 'A', 'c.1993G>A', 'FERMT1', 'p.D665N')\n",
      "❌ No ClinVar match for GAR1 p.R13Q\n",
      "❌ No ClinVar match for GK2 p.N277Tfs*25\n",
      "❌ No ClinVar match for GOLGA5 p.Q618*\n",
      "❌ No ClinVar match for GPR32 p.R355Gfs*16\n",
      "❌ No ClinVar match for IGF2R p.E2412K\n",
      "('7', '16308536', '16308536', '.', 'C', 'T', 'c.776C>T', 'ISPD', 'p.P259L')\n",
      "❌ No ClinVar match for KIAA1586 p.E378K\n",
      "❌ No ClinVar match for MORC1 p.Q857H\n",
      "❌ No ClinVar match for NCOR1 p.S174L\n",
      "❌ No ClinVar match for RGS4 p.R269Q\n",
      "('4', '989096', '989096', '.', 'G', 'A', 'c.1843G>A', 'SLC26A1', 'p.D615N')\n",
      "❌ No ClinVar match for SP110 p.D686Y\n",
      "❌ No ClinVar match for T p.R169*\n",
      "❌ No ClinVar match for TOP1 p.D301N\n",
      "('17', '7674252', '7674252', '.', 'G', 'C', 'c.711G>C', 'TP53', 'p.M237I')\n",
      "❌ No ClinVar match for TRRAP p.K2344Rfs*8p.K2376N\n",
      "❌ No ClinVar match for WDR5B p.E54K\n",
      "❌ No ClinVar match for ZNF71 p.V105I\n",
      "('6', '52982621', '52982621', '.', 'C', 'G', 'c.499C>G', 'GSTA4', '.')\n",
      "❌ No ClinVar match for SNX20 p.P35L\n",
      "❌ No ClinVar match for HSPA4 p.N799Mfs*50\n",
      "❌ No ClinVar match for TAB1 p.E356K\n",
      "❌ No ClinVar match for GRM1 p.T363K\n",
      "❌ No ClinVar match for PAPLN p.P1006A\n",
      "❌ No ClinVar match for PDCD10 p.Q155H\n",
      "('11', '103286310', '103286310', '.', 'G', 'A', 'c.10946G>A', 'DYNC2H1', 'p.R3656H')\n",
      "❌ No ClinVar match for SCAF1 p.E729G\n",
      "('22', '36312212', '36312212', '.', 'C', 'T', 'c.1565C>T', 'MYH9', 'p.P522L')\n",
      "('3', '36899201', '36899201', '.', 'G', 'A', 'c.341G>A', 'TRANK1', 'p.R70H')\n",
      "❌ No ClinVar match for INTS1 p.S1486N\n",
      "('19', '49104729', '49104729', '.', 'C', 'T', 'c.571C>T', 'SNRNP70', 'p.R191W')\n",
      "❌ No ClinVar match for ZMYND8 p.A931D\n",
      "('12', '25245350', '25245350', '.', 'G', 'A', 'c.35G>A', 'KRAS', 'p.G12D')\n",
      "❌ No ClinVar match for PDZRN3 p.T797A\n",
      "❌ No ClinVar match for BCL6 p.S128G\n",
      "❌ No ClinVar match for EBAG9 p.E114*\n",
      "❌ No ClinVar match for TPP1 p.H166Mfs*17\n",
      "❌ No ClinVar match for APC p.K1310Dfs*4p.R213*\n",
      "❌ No ClinVar match for ASXL3 p.P1943S\n",
      "✅ VCF written to clinvar_variants.vcf\n"
     ]
    }
   ],
   "execution_count": 233
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T12:16:01.876636Z",
     "start_time": "2025-04-22T12:16:01.843573Z"
    }
   },
   "cell_type": "code",
   "source": [
    "clinvar_vcf = pd.DataFrame([r.split('\\t') for r in vcf_lines[2:]], columns=[vcf_lines[1].split('\\t')])\n",
    "clinvar_vcf"
   ],
   "id": "bfc63cc3e16a1427",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "   #CHROM        POS       STOP ID REF ALT        cdna aa_change      gene  \\\n",
       "0      16   68822082   68822082  .   G   A   c.1793G>A      CDH1   p.R598Q   \n",
       "1      20    6077214    6077214  .   G   A   c.1993G>A    FERMT1   p.D665N   \n",
       "2       7   16308536   16308536  .   C   T    c.776C>T      ISPD   p.P259L   \n",
       "3       4     989096     989096  .   G   A   c.1843G>A   SLC26A1   p.D615N   \n",
       "4      17    7674252    7674252  .   G   C    c.711G>C      TP53   p.M237I   \n",
       "5       6   52982621   52982621  .   C   G    c.499C>G     GSTA4         .   \n",
       "6      11  103286310  103286310  .   G   A  c.10946G>A   DYNC2H1  p.R3656H   \n",
       "7      22   36312212   36312212  .   C   T   c.1565C>T      MYH9   p.P522L   \n",
       "8       3   36899201   36899201  .   G   A    c.341G>A    TRANK1    p.R70H   \n",
       "9      19   49104729   49104729  .   C   T    c.571C>T   SNRNP70   p.R191W   \n",
       "10     12   25245350   25245350  .   G   A     c.35G>A      KRAS    p.G12D   \n",
       "\n",
       "   QUAL FILTER INFO  \n",
       "0     .      .    .  \n",
       "1     .      .    .  \n",
       "2     .      .    .  \n",
       "3     .      .    .  \n",
       "4     .      .    .  \n",
       "5     .      .    .  \n",
       "6     .      .    .  \n",
       "7     .      .    .  \n",
       "8     .      .    .  \n",
       "9     .      .    .  \n",
       "10    .      .    .  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>STOP</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>cdna</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16</td>\n",
       "      <td>68822082</td>\n",
       "      <td>68822082</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.1793G&gt;A</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>p.R598Q</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>6077214</td>\n",
       "      <td>6077214</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.1993G&gt;A</td>\n",
       "      <td>FERMT1</td>\n",
       "      <td>p.D665N</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>16308536</td>\n",
       "      <td>16308536</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>c.776C&gt;T</td>\n",
       "      <td>ISPD</td>\n",
       "      <td>p.P259L</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>989096</td>\n",
       "      <td>989096</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.1843G&gt;A</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>p.D615N</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17</td>\n",
       "      <td>7674252</td>\n",
       "      <td>7674252</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>c.711G&gt;C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>p.M237I</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>52982621</td>\n",
       "      <td>52982621</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>c.499C&gt;G</td>\n",
       "      <td>GSTA4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>11</td>\n",
       "      <td>103286310</td>\n",
       "      <td>103286310</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.10946G&gt;A</td>\n",
       "      <td>DYNC2H1</td>\n",
       "      <td>p.R3656H</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>22</td>\n",
       "      <td>36312212</td>\n",
       "      <td>36312212</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>c.1565C&gt;T</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>p.P522L</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3</td>\n",
       "      <td>36899201</td>\n",
       "      <td>36899201</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.341G&gt;A</td>\n",
       "      <td>TRANK1</td>\n",
       "      <td>p.R70H</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>19</td>\n",
       "      <td>49104729</td>\n",
       "      <td>49104729</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>c.571C&gt;T</td>\n",
       "      <td>SNRNP70</td>\n",
       "      <td>p.R191W</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>25245350</td>\n",
       "      <td>25245350</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.35G&gt;A</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>p.G12D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 245
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T13:56:08.937159Z",
     "start_time": "2025-04-22T13:56:08.737500Z"
    }
   },
   "cell_type": "code",
   "source": "final_mut_df.reset_index(drop=True).merge(clinvar_vcf.reset_index(drop=True), left_on=\"symbol\", right_on=\"aa_change\")",
   "id": "1bf09ecf4ed45f10",
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "The column label 'aa_change' is not unique.\nFor a multi-index, the label must be a tuple with elements corresponding to each level.",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mValueError\u001B[0m                                Traceback (most recent call last)",
      "\u001B[0;32m/var/folders/_2/g5d3zf4s41g0tvlqkc3_8wjr0000gp/T/ipykernel_59856/1289668731.py\u001B[0m in \u001B[0;36m?\u001B[0;34m()\u001B[0m\n\u001B[0;32m----> 1\u001B[0;31m \u001B[0mfinal_mut_df\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mreset_index\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mdrop\u001B[0m\u001B[0;34m=\u001B[0m\u001B[0;32mTrue\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mmerge\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mclinvar_vcf\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mreset_index\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mdrop\u001B[0m\u001B[0;34m=\u001B[0m\u001B[0;32mTrue\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mleft_on\u001B[0m\u001B[0;34m=\u001B[0m\u001B[0;34m\"symbol\"\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mright_on\u001B[0m\u001B[0;34m=\u001B[0m\u001B[0;34m\"aa_change\"\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m",
      "\u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/pandas/core/frame.py\u001B[0m in \u001B[0;36m?\u001B[0;34m(self, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001B[0m\n\u001B[1;32m  10089\u001B[0m         \u001B[0mvalidate\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mstr\u001B[0m \u001B[0;34m|\u001B[0m \u001B[0;32mNone\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mNone\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m  10090\u001B[0m     ) -> DataFrame:\n\u001B[1;32m  10091\u001B[0m         \u001B[0;32mfrom\u001B[0m \u001B[0mpandas\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mcore\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mreshape\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mmerge\u001B[0m \u001B[0;32mimport\u001B[0m \u001B[0mmerge\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m  10092\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m> 10093\u001B[0;31m         return merge(\n\u001B[0m\u001B[1;32m  10094\u001B[0m             \u001B[0mself\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m  10095\u001B[0m             \u001B[0mright\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m  10096\u001B[0m             \u001B[0mhow\u001B[0m\u001B[0;34m=\u001B[0m\u001B[0mhow\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n",
      "\u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/pandas/core/reshape/merge.py\u001B[0m in \u001B[0;36m?\u001B[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001B[0m\n\u001B[1;32m    106\u001B[0m     \u001B[0mcopy\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mbool\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mTrue\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    107\u001B[0m     \u001B[0mindicator\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mbool\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mFalse\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    108\u001B[0m     \u001B[0mvalidate\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mstr\u001B[0m \u001B[0;34m|\u001B[0m \u001B[0;32mNone\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mNone\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    109\u001B[0m ) -> DataFrame:\n\u001B[0;32m--> 110\u001B[0;31m     op = _MergeOperation(\n\u001B[0m\u001B[1;32m    111\u001B[0m         \u001B[0mleft\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    112\u001B[0m         \u001B[0mright\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    113\u001B[0m         \u001B[0mhow\u001B[0m\u001B[0;34m=\u001B[0m\u001B[0mhow\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n",
      "\u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/pandas/core/reshape/merge.py\u001B[0m in \u001B[0;36m?\u001B[0;34m(self, left, right, how, on, left_on, right_on, axis, left_index, right_index, sort, suffixes, indicator, validate)\u001B[0m\n\u001B[1;32m    699\u001B[0m         (\n\u001B[1;32m    700\u001B[0m             \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mleft_join_keys\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    701\u001B[0m             \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mright_join_keys\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    702\u001B[0m             \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mjoin_names\u001B[0m\u001B[0;34m,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m--> 703\u001B[0;31m         ) = self._get_merge_keys()\n\u001B[0m\u001B[1;32m    704\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    705\u001B[0m         \u001B[0;31m# validate the merge keys dtypes. We may need to coerce\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    706\u001B[0m         \u001B[0;31m# to avoid incompatible dtypes\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n",
      "\u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/pandas/core/reshape/merge.py\u001B[0m in \u001B[0;36m?\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m   1158\u001B[0m                         \u001B[0;31m# Then we're either Hashable or a wrong-length arraylike,\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1159\u001B[0m                         \u001B[0;31m#  the latter of which will raise\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1160\u001B[0m                         \u001B[0mrk\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mcast\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mHashable\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mrk\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1161\u001B[0m                         \u001B[0;32mif\u001B[0m \u001B[0mrk\u001B[0m \u001B[0;32mis\u001B[0m \u001B[0;32mnot\u001B[0m \u001B[0;32mNone\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m-> 1162\u001B[0;31m                             \u001B[0mright_keys\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mappend\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mright\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_get_label_or_level_values\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mrk\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m   1163\u001B[0m                         \u001B[0;32melse\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1164\u001B[0m                             \u001B[0;31m# work-around for merge_asof(right_index=True)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1165\u001B[0m                             \u001B[0mright_keys\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mappend\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mright\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mindex\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n",
      "\u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/pandas/core/generic.py\u001B[0m in \u001B[0;36m?\u001B[0;34m(self, key, axis)\u001B[0m\n\u001B[1;32m   1861\u001B[0m             \u001B[0;32melse\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1862\u001B[0m                 \u001B[0mmulti_message\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;34m\"\"\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1863\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1864\u001B[0m             \u001B[0mlabel_axis_name\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;34m\"column\"\u001B[0m \u001B[0;32mif\u001B[0m \u001B[0maxis\u001B[0m \u001B[0;34m==\u001B[0m \u001B[0;36m0\u001B[0m \u001B[0;32melse\u001B[0m \u001B[0;34m\"index\"\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m-> 1865\u001B[0;31m             raise ValueError(\n\u001B[0m\u001B[1;32m   1866\u001B[0m                 \u001B[0;34mf\"The {label_axis_name} label '{key}' is not unique.{multi_message}\"\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   1867\u001B[0m             )\n\u001B[1;32m   1868\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n",
      "\u001B[0;31mValueError\u001B[0m: The column label 'aa_change' is not unique.\nFor a multi-index, the label must be a tuple with elements corresponding to each level."
     ]
    }
   ],
   "execution_count": 267
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T12:21:54.551066Z",
     "start_time": "2025-04-22T12:21:54.526247Z"
    }
   },
   "cell_type": "code",
   "source": "final_mut_df[final_mut_df['symbol'] == 'GSTA4']",
   "id": "5ac2fc27e59b2e98",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            sample_id symbol read_depth allele_frequency chromosome strand  \\\n",
       "4090  WU-L012-T1-X1P0  GSTA4        NaN            0.312        NaN    NaN   \n",
       "\n",
       "     seq_start_position seq_end_position ref_allele alt_allele variation_id  \\\n",
       "4090                NaN              NaN        NaN        NaN          NaN   \n",
       "\n",
       "     platform amino_acid_change  consequence variant_class     platform_id  \n",
       "4090      NaN                 .  Splice_Site           DEL  wustl_mutation  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>read_depth</th>\n",
       "      <th>allele_frequency</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>seq_start_position</th>\n",
       "      <th>seq_end_position</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>platform</th>\n",
       "      <th>amino_acid_change</th>\n",
       "      <th>consequence</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>platform_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4090</th>\n",
       "      <td>WU-L012-T1-X1P0</td>\n",
       "      <td>GSTA4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>DEL</td>\n",
       "      <td>wustl_mutation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 262
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T12:21:45.258283Z",
     "start_time": "2025-04-22T12:21:45.244019Z"
    }
   },
   "cell_type": "code",
   "source": "clinvar_vcf['gene']",
   "id": "e505e2917472d077",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        gene\n",
       "0    p.R598Q\n",
       "1    p.D665N\n",
       "2    p.P259L\n",
       "3    p.D615N\n",
       "4    p.M237I\n",
       "5          .\n",
       "6   p.R3656H\n",
       "7    p.P522L\n",
       "8     p.R70H\n",
       "9    p.R191W\n",
       "10    p.G12D"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>p.R598Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>p.D665N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>p.P259L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>p.D615N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>p.M237I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>p.R3656H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>p.P522L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>p.R70H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>p.R191W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>p.G12D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 261
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:51:52.453851Z",
     "start_time": "2025-04-22T10:51:51.745964Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import requests\n",
    "\n",
    "def get_hgnc_symbol(symbol):\n",
    "    url = \"https://mygene.info/v3/query\"\n",
    "    params = {\n",
    "        \"q\": symbol,\n",
    "        \"species\": \"human\",\n",
    "        \"fields\": \"symbol,name,HGNC\",\n",
    "        \"size\": 2\n",
    "    }\n",
    "    response = requests.get(url, params=params)\n",
    "    data = response.json()\n",
    "    if data[\"hits\"]:\n",
    "        return data['hits']\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "# Example usage\n",
    "data = get_hgnc_symbol(\"Tp53\")   # returns \"TP53\"\n",
    "#print(get_hgnc_symbol(\"P53\"))    # returns \"TP53\" (aliases work too)\n"
   ],
   "id": "742a6baf3ef1271c",
   "outputs": [],
   "execution_count": 223
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:52:06.695416Z",
     "start_time": "2025-04-22T10:52:06.688933Z"
    }
   },
   "cell_type": "code",
   "source": "data[1]",
   "id": "f76eb67138f9fd8d",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HGNC': '44657',\n",
       " '_id': '729264',\n",
       " '_score': 63.653282,\n",
       " 'name': 'TP53 target 3D',\n",
       " 'symbol': 'TP53TG3D'}"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 226
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:12:20.053677Z",
     "start_time": "2025-04-22T10:12:19.573350Z"
    }
   },
   "cell_type": "code",
   "source": "record = fetch_clinvar_record('188221')",
   "id": "3f47217e8e3dc6b6",
   "outputs": [],
   "execution_count": 179
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:36:31.335593Z",
     "start_time": "2025-04-22T10:36:31.324495Z"
    }
   },
   "cell_type": "code",
   "source": "record['variation_set'][0]",
   "id": "9f1e602ec903baa5",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'measure_id': '27621',\n",
       " 'variation_xrefs': [{'db_source': 'ClinGen', 'db_id': 'CA122538'},\n",
       "  {'db_source': 'UniProtKB', 'db_id': 'P01116#VAR_016026'},\n",
       "  {'db_source': 'OMIM', 'db_id': '190070.0005'},\n",
       "  {'db_source': 'OMIM', 'db_id': '190070.0025'},\n",
       "  {'db_source': 'dbSNP', 'db_id': '121913529'}],\n",
       " 'variation_name': 'NM_004985.5(KRAS):c.35G>A (p.Gly12Asp)',\n",
       " 'cdna_change': 'c.35G>A',\n",
       " 'aliases': [],\n",
       " 'variation_loc': [{'status': 'current',\n",
       "   'assembly_name': 'GRCh38',\n",
       "   'chr': '12',\n",
       "   'band': '12p12.1',\n",
       "   'start': '25245350',\n",
       "   'stop': '25245350',\n",
       "   'inner_start': '',\n",
       "   'inner_stop': '',\n",
       "   'outer_start': '',\n",
       "   'outer_stop': '',\n",
       "   'display_start': '25245350',\n",
       "   'display_stop': '25245350',\n",
       "   'assembly_acc_ver': 'GCF_000001405.38',\n",
       "   'annotation_release': '',\n",
       "   'alt': '',\n",
       "   'ref': ''},\n",
       "  {'status': 'previous',\n",
       "   'assembly_name': 'GRCh37',\n",
       "   'chr': '12',\n",
       "   'band': '12p12.1',\n",
       "   'start': '25398284',\n",
       "   'stop': '25398284',\n",
       "   'inner_start': '',\n",
       "   'inner_stop': '',\n",
       "   'outer_start': '',\n",
       "   'outer_stop': '',\n",
       "   'display_start': '25398284',\n",
       "   'display_stop': '25398284',\n",
       "   'assembly_acc_ver': 'GCF_000001405.25',\n",
       "   'annotation_release': '',\n",
       "   'alt': '',\n",
       "   'ref': ''}],\n",
       " 'allele_freq_set': [{'source': 'The Genome Aggregation Database (gnomAD), exomes',\n",
       "   'value': '0.00000',\n",
       "   'minor_allele': ''},\n",
       "  {'source': 'The Genome Aggregation Database (gnomAD)',\n",
       "   'value': '0.00001',\n",
       "   'minor_allele': ''},\n",
       "  {'source': 'Exome Aggregation Consortium (ExAC)',\n",
       "   'value': '0.00002',\n",
       "   'minor_allele': ''}],\n",
       " 'variant_type': 'single nucleotide variant',\n",
       " 'canonical_spdi': 'NC_000012.12:25245349:C:T',\n",
       " 'common_name': ''}"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 202
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T10:13:28.532003Z",
     "start_time": "2025-04-22T10:13:28.526880Z"
    }
   },
   "cell_type": "code",
   "source": "extract_vcf_info(record)",
   "id": "6c2fa2fa48250abf",
   "outputs": [],
   "execution_count": 181
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T10:11:05.429845Z",
     "start_time": "2025-04-16T10:11:05.409229Z"
    }
   },
   "cell_type": "code",
   "source": "mms",
   "id": "15ffad810acbe847",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                   Field                                           model_id  \\\n",
       "0           #Description  Unique identifier for all the PDXs derived fro...   \n",
       "1               #Example                                          CRC0228PR   \n",
       "2   #Format Requirements                                free alphanumerical   \n",
       "3            #Essential?                                          essential   \n",
       "4                    NaN                                        WUSTL WHIM5   \n",
       "5                    NaN                                        WUSTL WHIM6   \n",
       "6                    NaN                                        WUSTL WHIM8   \n",
       "7                    NaN                                        WUSTL WHIM9   \n",
       "8                    NaN                                       WUSTL WHIM11   \n",
       "9                    NaN                                       WUSTL WHIM12   \n",
       "10                   NaN                                       WUSTL WHIM13   \n",
       "11                   NaN                                       WUSTL WHIM14   \n",
       "12                   NaN                                       WUSTL WHIM16   \n",
       "13                   NaN                                       WUSTL WHIM18   \n",
       "14                   NaN                                       WUSTL WHIM20   \n",
       "15                   NaN                                       WUSTL WHIM21   \n",
       "\n",
       "                                            sample_id  \\\n",
       "0   Identifier of the sample from any patient tiss...   \n",
       "1                          CRC0228PRH0000000000D01000   \n",
       "2                                 free alphanumerical   \n",
       "3                                           essential   \n",
       "4                                               WHIM5   \n",
       "5                                               WHIM6   \n",
       "6                                               WHIM8   \n",
       "7                                               WHIM9   \n",
       "8                                              WHIM11   \n",
       "9                                              WHIM12   \n",
       "10                                             WHIM13   \n",
       "11                                             WHIM14   \n",
       "12                                             WHIM16   \n",
       "13                                             WHIM18   \n",
       "14                                             WHIM20   \n",
       "15                                             WHIM21   \n",
       "\n",
       "                                        sample_origin  \\\n",
       "0   The orgin of the sample - xenograft (extracted...   \n",
       "1                                           xenograft   \n",
       "2                                  xenograft, patient   \n",
       "3                                           essential   \n",
       "4                                           xenograft   \n",
       "5                                           xenograft   \n",
       "6                                           xenograft   \n",
       "7                                           xenograft   \n",
       "8                                           xenograft   \n",
       "9                                           xenograft   \n",
       "10                                          xenograft   \n",
       "11                                          xenograft   \n",
       "12                                          xenograft   \n",
       "13                                          xenograft   \n",
       "14                                          xenograft   \n",
       "15                                          xenograft   \n",
       "\n",
       "                                              passage  \\\n",
       "0   Indicate the passage number of the sample wher...   \n",
       "1                                                   0   \n",
       "2                                 numerical, or blank   \n",
       "3                                           essential   \n",
       "4                                                   0   \n",
       "5                                                   0   \n",
       "6                                                   0   \n",
       "7                                                   0   \n",
       "8                                                   0   \n",
       "9                                                   0   \n",
       "10                                                  0   \n",
       "11                                                  0   \n",
       "12                                                  0   \n",
       "13                                                  0   \n",
       "14                                                  0   \n",
       "15                                                  0   \n",
       "\n",
       "                                     host_strain_name  \\\n",
       "0   Host mouse strain name (e.g. NOD-SCID, NSG, et...   \n",
       "1                                            NOD SCID   \n",
       "2    follow strain name or Not Specified (if unknown)   \n",
       "3                                           essential   \n",
       "4                                                 NSG   \n",
       "5                                                 NSG   \n",
       "6                                                 NSG   \n",
       "7                                                 NSG   \n",
       "8                                                 NSG   \n",
       "9                                                 NSG   \n",
       "10                                                NSG   \n",
       "11                                                NSG   \n",
       "12                                                NSG   \n",
       "13                                                NSG   \n",
       "14                                                NSG   \n",
       "15                                                NSG   \n",
       "\n",
       "                             host_strain_nomenclature  \\\n",
       "0   The full nomenclature form of the host mouse s...   \n",
       "1                    NOD.Cg-Prkdcscid Il2rgtm1Wjl/SzJ   \n",
       "2   https://www.jax.org/jax-mice-and-services/cust...   \n",
       "3                                           essential   \n",
       "4   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "5   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "6   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "7   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "8   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "9   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "10  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "11  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "12  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "13  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "14  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "15  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "\n",
       "                      engrafted_tumor_collection_site  \\\n",
       "0   The anatomical site from which the xenograft s...   \n",
       "1                                        mammary glad   \n",
       "2                                 free alphanumerical   \n",
       "3                                           desirable   \n",
       "4                                     mammary fat pad   \n",
       "5                                     mammary fat pad   \n",
       "6                                     mammary fat pad   \n",
       "7                                     mammary fat pad   \n",
       "8                                     mammary fat pad   \n",
       "9                                     mammary fat pad   \n",
       "10                                    mammary fat pad   \n",
       "11                                    mammary fat pad   \n",
       "12                                    mammary fat pad   \n",
       "13                                    mammary fat pad   \n",
       "14                                    mammary fat pad   \n",
       "15                                    mammary fat pad   \n",
       "\n",
       "                                         raw_data_url  \\\n",
       "0   Unique identify to platform metadata between s...   \n",
       "1                                           mutation1   \n",
       "2                                                 NaN   \n",
       "3                                           desirable   \n",
       "4                   phs000611,  GSE46604, PRJNA217715   \n",
       "5                   phs000611,  GSE46604, PRJNA217715   \n",
       "6                   phs000611,  GSE46604, PRJNA217715   \n",
       "7                   phs000611,  GSE46604, PRJNA217715   \n",
       "8                   phs000611,  GSE46604, PRJNA217715   \n",
       "9                   phs000611,  GSE46604, PRJNA217715   \n",
       "10                  phs000611,  GSE46604, PRJNA217715   \n",
       "11                  phs000611,  GSE46604, PRJNA217715   \n",
       "12                  phs000611,  GSE46604, PRJNA217715   \n",
       "13                  phs000611,  GSE46604, PRJNA217715   \n",
       "14                  phs000611,  GSE46604, PRJNA217715   \n",
       "15                  phs000611,  GSE46604, PRJNA217715   \n",
       "\n",
       "                                          platform_id  \n",
       "0   Unique identify to platform metadata between s...  \n",
       "1                                           mutation1  \n",
       "2                                                 NaN  \n",
       "3                                           essential  \n",
       "4                                    mut_wgs_illumina  \n",
       "5                                    mut_wgs_illumina  \n",
       "6                                    mut_wgs_illumina  \n",
       "7                                    mut_wgs_illumina  \n",
       "8                                    mut_wgs_illumina  \n",
       "9                                    mut_wgs_illumina  \n",
       "10                                   mut_wgs_illumina  \n",
       "11                                   mut_wgs_illumina  \n",
       "12                                   mut_wgs_illumina  \n",
       "13                                   mut_wgs_illumina  \n",
       "14                                   mut_wgs_illumina  \n",
       "15                                   mut_wgs_illumina  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Field</th>\n",
       "      <th>model_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>sample_origin</th>\n",
       "      <th>passage</th>\n",
       "      <th>host_strain_name</th>\n",
       "      <th>host_strain_nomenclature</th>\n",
       "      <th>engrafted_tumor_collection_site</th>\n",
       "      <th>raw_data_url</th>\n",
       "      <th>platform_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>#Description</td>\n",
       "      <td>Unique identifier for all the PDXs derived fro...</td>\n",
       "      <td>Identifier of the sample from any patient tiss...</td>\n",
       "      <td>The orgin of the sample - xenograft (extracted...</td>\n",
       "      <td>Indicate the passage number of the sample wher...</td>\n",
       "      <td>Host mouse strain name (e.g. NOD-SCID, NSG, et...</td>\n",
       "      <td>The full nomenclature form of the host mouse s...</td>\n",
       "      <td>The anatomical site from which the xenograft s...</td>\n",
       "      <td>Unique identify to platform metadata between s...</td>\n",
       "      <td>Unique identify to platform metadata between s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#Example</td>\n",
       "      <td>CRC0228PR</td>\n",
       "      <td>CRC0228PRH0000000000D01000</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.Cg-Prkdcscid Il2rgtm1Wjl/SzJ</td>\n",
       "      <td>mammary glad</td>\n",
       "      <td>mutation1</td>\n",
       "      <td>mutation1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>#Format Requirements</td>\n",
       "      <td>free alphanumerical</td>\n",
       "      <td>free alphanumerical</td>\n",
       "      <td>xenograft, patient</td>\n",
       "      <td>numerical, or blank</td>\n",
       "      <td>follow strain name or Not Specified (if unknown)</td>\n",
       "      <td>https://www.jax.org/jax-mice-and-services/cust...</td>\n",
       "      <td>free alphanumerical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>#Essential?</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>desirable</td>\n",
       "      <td>desirable</td>\n",
       "      <td>essential</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM5</td>\n",
       "      <td>WHIM5</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM6</td>\n",
       "      <td>WHIM6</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM8</td>\n",
       "      <td>WHIM8</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM9</td>\n",
       "      <td>WHIM9</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM11</td>\n",
       "      <td>WHIM11</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM12</td>\n",
       "      <td>WHIM12</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM13</td>\n",
       "      <td>WHIM13</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM14</td>\n",
       "      <td>WHIM14</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM16</td>\n",
       "      <td>WHIM16</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM18</td>\n",
       "      <td>WHIM18</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM20</td>\n",
       "      <td>WHIM20</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM21</td>\n",
       "      <td>WHIM21</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 132
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "b5401f47ebba6bf2"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T10:14:07.678535Z",
     "start_time": "2025-04-16T10:14:07.662709Z"
    }
   },
   "cell_type": "code",
   "source": "mms",
   "id": "371206111096b9ed",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                   Field                                           model_id  \\\n",
       "0           #Description  Unique identifier for all the PDXs derived fro...   \n",
       "1               #Example                                          CRC0228PR   \n",
       "2   #Format Requirements                                free alphanumerical   \n",
       "3            #Essential?                                          essential   \n",
       "4                    NaN                                        WUSTL WHIM5   \n",
       "5                    NaN                                        WUSTL WHIM6   \n",
       "6                    NaN                                        WUSTL WHIM8   \n",
       "7                    NaN                                        WUSTL WHIM9   \n",
       "8                    NaN                                       WUSTL WHIM11   \n",
       "9                    NaN                                       WUSTL WHIM12   \n",
       "10                   NaN                                       WUSTL WHIM13   \n",
       "11                   NaN                                       WUSTL WHIM14   \n",
       "12                   NaN                                       WUSTL WHIM16   \n",
       "13                   NaN                                       WUSTL WHIM18   \n",
       "14                   NaN                                       WUSTL WHIM20   \n",
       "15                   NaN                                       WUSTL WHIM21   \n",
       "16                                                               WU-0029-T1   \n",
       "17                                                               WU-0029-T1   \n",
       "\n",
       "                                            sample_id  \\\n",
       "0   Identifier of the sample from any patient tiss...   \n",
       "1                          CRC0228PRH0000000000D01000   \n",
       "2                                 free alphanumerical   \n",
       "3                                           essential   \n",
       "4                                               WHIM5   \n",
       "5                                               WHIM6   \n",
       "6                                               WHIM8   \n",
       "7                                               WHIM9   \n",
       "8                                              WHIM11   \n",
       "9                                              WHIM12   \n",
       "10                                             WHIM13   \n",
       "11                                             WHIM14   \n",
       "12                                             WHIM16   \n",
       "13                                             WHIM18   \n",
       "14                                             WHIM20   \n",
       "15                                             WHIM21   \n",
       "16                                    WU-0029-T1-X1P0   \n",
       "17                                    WU-0029-T1-X1P1   \n",
       "\n",
       "                                        sample_origin  \\\n",
       "0   The orgin of the sample - xenograft (extracted...   \n",
       "1                                           xenograft   \n",
       "2                                  xenograft, patient   \n",
       "3                                           essential   \n",
       "4                                           xenograft   \n",
       "5                                           xenograft   \n",
       "6                                           xenograft   \n",
       "7                                           xenograft   \n",
       "8                                           xenograft   \n",
       "9                                           xenograft   \n",
       "10                                          xenograft   \n",
       "11                                          xenograft   \n",
       "12                                          xenograft   \n",
       "13                                          xenograft   \n",
       "14                                          xenograft   \n",
       "15                                          xenograft   \n",
       "16                                          xenograft   \n",
       "17                                          xenograft   \n",
       "\n",
       "                                              passage  \\\n",
       "0   Indicate the passage number of the sample wher...   \n",
       "1                                                   0   \n",
       "2                                 numerical, or blank   \n",
       "3                                           essential   \n",
       "4                                                   0   \n",
       "5                                                   0   \n",
       "6                                                   0   \n",
       "7                                                   0   \n",
       "8                                                   0   \n",
       "9                                                   0   \n",
       "10                                                  0   \n",
       "11                                                  0   \n",
       "12                                                  0   \n",
       "13                                                  0   \n",
       "14                                                  0   \n",
       "15                                                  0   \n",
       "16                                                  0   \n",
       "17                                                  0   \n",
       "\n",
       "                                     host_strain_name  \\\n",
       "0   Host mouse strain name (e.g. NOD-SCID, NSG, et...   \n",
       "1                                            NOD SCID   \n",
       "2    follow strain name or Not Specified (if unknown)   \n",
       "3                                           essential   \n",
       "4                                                 NSG   \n",
       "5                                                 NSG   \n",
       "6                                                 NSG   \n",
       "7                                                 NSG   \n",
       "8                                                 NSG   \n",
       "9                                                 NSG   \n",
       "10                                                NSG   \n",
       "11                                                NSG   \n",
       "12                                                NSG   \n",
       "13                                                NSG   \n",
       "14                                                NSG   \n",
       "15                                                NSG   \n",
       "16                                                NSG   \n",
       "17                                                NSG   \n",
       "\n",
       "                             host_strain_nomenclature  \\\n",
       "0   The full nomenclature form of the host mouse s...   \n",
       "1                    NOD.Cg-Prkdcscid Il2rgtm1Wjl/SzJ   \n",
       "2   https://www.jax.org/jax-mice-and-services/cust...   \n",
       "3                                           essential   \n",
       "4   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "5   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "6   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "7   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "8   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "9   NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "10  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "11  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "12  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "13  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "14  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "15  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "16  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "17  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...   \n",
       "\n",
       "                      engrafted_tumor_collection_site  \\\n",
       "0   The anatomical site from which the xenograft s...   \n",
       "1                                        mammary glad   \n",
       "2                                 free alphanumerical   \n",
       "3                                           desirable   \n",
       "4                                     mammary fat pad   \n",
       "5                                     mammary fat pad   \n",
       "6                                     mammary fat pad   \n",
       "7                                     mammary fat pad   \n",
       "8                                     mammary fat pad   \n",
       "9                                     mammary fat pad   \n",
       "10                                    mammary fat pad   \n",
       "11                                    mammary fat pad   \n",
       "12                                    mammary fat pad   \n",
       "13                                    mammary fat pad   \n",
       "14                                    mammary fat pad   \n",
       "15                                    mammary fat pad   \n",
       "16                                                      \n",
       "17                                                      \n",
       "\n",
       "                                         raw_data_url  \\\n",
       "0   Unique identify to platform metadata between s...   \n",
       "1                                           mutation1   \n",
       "2                                                 NaN   \n",
       "3                                           desirable   \n",
       "4                   phs000611,  GSE46604, PRJNA217715   \n",
       "5                   phs000611,  GSE46604, PRJNA217715   \n",
       "6                   phs000611,  GSE46604, PRJNA217715   \n",
       "7                   phs000611,  GSE46604, PRJNA217715   \n",
       "8                   phs000611,  GSE46604, PRJNA217715   \n",
       "9                   phs000611,  GSE46604, PRJNA217715   \n",
       "10                  phs000611,  GSE46604, PRJNA217715   \n",
       "11                  phs000611,  GSE46604, PRJNA217715   \n",
       "12                  phs000611,  GSE46604, PRJNA217715   \n",
       "13                  phs000611,  GSE46604, PRJNA217715   \n",
       "14                  phs000611,  GSE46604, PRJNA217715   \n",
       "15                  phs000611,  GSE46604, PRJNA217715   \n",
       "16                                                      \n",
       "17                                                      \n",
       "\n",
       "                                          platform_id  \n",
       "0   Unique identify to platform metadata between s...  \n",
       "1                                           mutation1  \n",
       "2                                                 NaN  \n",
       "3                                           essential  \n",
       "4                                    mut_wgs_illumina  \n",
       "5                                    mut_wgs_illumina  \n",
       "6                                    mut_wgs_illumina  \n",
       "7                                    mut_wgs_illumina  \n",
       "8                                    mut_wgs_illumina  \n",
       "9                                    mut_wgs_illumina  \n",
       "10                                   mut_wgs_illumina  \n",
       "11                                   mut_wgs_illumina  \n",
       "12                                   mut_wgs_illumina  \n",
       "13                                   mut_wgs_illumina  \n",
       "14                                   mut_wgs_illumina  \n",
       "15                                   mut_wgs_illumina  \n",
       "16                                   wustl_cna_status  \n",
       "17                                   wustl_cna_status  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Field</th>\n",
       "      <th>model_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>sample_origin</th>\n",
       "      <th>passage</th>\n",
       "      <th>host_strain_name</th>\n",
       "      <th>host_strain_nomenclature</th>\n",
       "      <th>engrafted_tumor_collection_site</th>\n",
       "      <th>raw_data_url</th>\n",
       "      <th>platform_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>#Description</td>\n",
       "      <td>Unique identifier for all the PDXs derived fro...</td>\n",
       "      <td>Identifier of the sample from any patient tiss...</td>\n",
       "      <td>The orgin of the sample - xenograft (extracted...</td>\n",
       "      <td>Indicate the passage number of the sample wher...</td>\n",
       "      <td>Host mouse strain name (e.g. NOD-SCID, NSG, et...</td>\n",
       "      <td>The full nomenclature form of the host mouse s...</td>\n",
       "      <td>The anatomical site from which the xenograft s...</td>\n",
       "      <td>Unique identify to platform metadata between s...</td>\n",
       "      <td>Unique identify to platform metadata between s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#Example</td>\n",
       "      <td>CRC0228PR</td>\n",
       "      <td>CRC0228PRH0000000000D01000</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.Cg-Prkdcscid Il2rgtm1Wjl/SzJ</td>\n",
       "      <td>mammary glad</td>\n",
       "      <td>mutation1</td>\n",
       "      <td>mutation1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>#Format Requirements</td>\n",
       "      <td>free alphanumerical</td>\n",
       "      <td>free alphanumerical</td>\n",
       "      <td>xenograft, patient</td>\n",
       "      <td>numerical, or blank</td>\n",
       "      <td>follow strain name or Not Specified (if unknown)</td>\n",
       "      <td>https://www.jax.org/jax-mice-and-services/cust...</td>\n",
       "      <td>free alphanumerical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>#Essential?</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>essential</td>\n",
       "      <td>desirable</td>\n",
       "      <td>desirable</td>\n",
       "      <td>essential</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM5</td>\n",
       "      <td>WHIM5</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM6</td>\n",
       "      <td>WHIM6</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM8</td>\n",
       "      <td>WHIM8</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM9</td>\n",
       "      <td>WHIM9</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM11</td>\n",
       "      <td>WHIM11</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM12</td>\n",
       "      <td>WHIM12</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM13</td>\n",
       "      <td>WHIM13</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM14</td>\n",
       "      <td>WHIM14</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM16</td>\n",
       "      <td>WHIM16</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM18</td>\n",
       "      <td>WHIM18</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM20</td>\n",
       "      <td>WHIM20</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>WUSTL WHIM21</td>\n",
       "      <td>WHIM21</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>phs000611,  GSE46604, PRJNA217715</td>\n",
       "      <td>mut_wgs_illumina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td></td>\n",
       "      <td>WU-0029-T1</td>\n",
       "      <td>WU-0029-T1-X1P0</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>wustl_cna_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td></td>\n",
       "      <td>WU-0029-T1</td>\n",
       "      <td>WU-0029-T1-X1P1</td>\n",
       "      <td>xenograft</td>\n",
       "      <td>0</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>wustl_cna_status</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 136
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:55:27.363835Z",
     "start_time": "2025-04-16T09:55:27.356752Z"
    }
   },
   "cell_type": "code",
   "source": "len(table)",
   "id": "a462a51a51a6a3c",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 120
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:58:51.311390Z",
     "start_time": "2025-04-16T09:58:51.302861Z"
    }
   },
   "cell_type": "code",
   "source": "table[0]",
   "id": "4df5ad02c07d343a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<table class=\"center\"><thead><tr class=\"gray-bg\"><th class=\"calign\">Variant Type</th><th class=\"calign\">Calling Status</th></tr></thead><tbody><tr><td>Mutations</td><td>Called</td></tr><tr><td>Copy number</td><td>Called</td></tr><tr><td>Fusions</td><td>Called</td></tr></tbody></table>"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 124
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:58:03.509860Z",
     "start_time": "2025-04-16T09:58:02.555585Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Extract headers\n",
    "headers = [th.get_text(strip=True).replace('\\n', ' ') for th in table[1].find_all('th')]\n",
    "\n",
    "# Extract rows\n",
    "rows = []\n",
    "for tr in soup.find_all('tr')[1:]:\n",
    "    cells = tr.find_all('td')\n",
    "    row = [td.get_text(strip=True).replace('\\n', '') for td in cells]\n",
    "    if row:\n",
    "        rows.append(row)\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(rows, columns=headers)\n"
   ],
   "id": "a03d14eaf79c7442",
   "outputs": [],
   "execution_count": 122
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-16T09:58:07.912416Z",
     "start_time": "2025-04-16T09:58:07.897888Z"
    }
   },
   "cell_type": "code",
   "source": "df",
   "id": "b633db78d14daf06",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             Gene SomaticMutations Classification  Type   VAF  \\\n",
       "0       Mutations           Called           None  None  None   \n",
       "1     Copy number           Called           None  None  None   \n",
       "2         Fusions           Called           None  None  None   \n",
       "3             A2M                -              -     -     -   \n",
       "4           A2ML1                -              -     -     -   \n",
       "...           ...              ...            ...   ...   ...   \n",
       "9157       ZSWIM6                -              -     -     -   \n",
       "9158       ZSWIM7                -              -     -     -   \n",
       "9159         ZW10                -              -     -     -   \n",
       "9160         ZXDC                -              -     -     -   \n",
       "9161        ZZEF1                -              -     -     -   \n",
       "\n",
       "     SomaticCopy Number CN status Fusions  \n",
       "0                  None      None    None  \n",
       "1                  None      None    None  \n",
       "2                  None      None    None  \n",
       "3                     5       Amp       -  \n",
       "4                     5       Amp       -  \n",
       "...                 ...       ...     ...  \n",
       "9157                  1      Loss       -  \n",
       "9158                  3      Gain       -  \n",
       "9159                  1      Loss       -  \n",
       "9160                  1      Loss       -  \n",
       "9161                  3      Gain       -  \n",
       "\n",
       "[9162 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>SomaticMutations</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Type</th>\n",
       "      <th>VAF</th>\n",
       "      <th>SomaticCopy Number</th>\n",
       "      <th>CN status</th>\n",
       "      <th>Fusions</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Mutations</td>\n",
       "      <td>Called</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Copy number</td>\n",
       "      <td>Called</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Fusions</td>\n",
       "      <td>Called</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>Amp</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>Amp</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9157</th>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Loss</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9158</th>\n",
       "      <td>ZSWIM7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>Gain</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9159</th>\n",
       "      <td>ZW10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Loss</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9160</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Loss</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9161</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>Gain</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9162 rows × 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 123
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T08:05:26.471214Z",
     "start_time": "2025-04-22T08:05:15.537706Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from utils import read_metadata_without_fields\n",
    "metadata = pd.DataFrame()\n",
    "for provider in get_dirs(home):\n",
    "    new_path = join(home, provider)\n",
    "    tsv_files = sorted([join(new_path, f) for f in get_files(new_path) if f.endswith('.tsv') and not f.__contains__('image') and not f.__contains__('molecular_metadata')])\n",
    "    if f'{home}{provider}/{provider}_metadata-pdx_model.tsv' in tsv_files:\n",
    "        df = read_metadata_without_fields(f'{home}{provider}/{provider}_metadata-pdx_model.tsv')\n",
    "        f =  f'{home}{provider}/{provider}_metadata-model_validation.tsv'\n",
    "        df = df.merge(read_metadata_without_fields(f), on='model_id', how='left')\n",
    "        f =  f'{home}{provider}/{provider}_metadata-patient_sample.tsv'\n",
    "        df = df.merge(read_metadata_without_fields(f), on='model_id', how='left')\n",
    "        f =  f'{home}{provider}/{provider}_metadata-patient.tsv'\n",
    "        df = df.merge(read_metadata_without_fields(f), on='patient_id', how='left')\n",
    "    metadata = pd.concat([metadata, df], ignore_index=True)\n",
    "\n",
    "metadata\n"
   ],
   "id": "81398879f13581fd",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             model_id host_strain_name  \\\n",
       "0       VHIO_PC-PDX-1         NOD SCID   \n",
       "1       VHIO_PC-PDX-3         NOD SCID   \n",
       "2       VHIO_PC-PDX-6         NOD SCID   \n",
       "3       VHIO_PC-PDX-9         NOD SCID   \n",
       "4      VHIO_PC-PDX-10         NOD SCID   \n",
       "...               ...              ...   \n",
       "14083            472M              NSG   \n",
       "14084            472M              NSG   \n",
       "14085         472M_Cx              NSG   \n",
       "14086         472M_Cx              NSG   \n",
       "14087         472M_Cx              NSG   \n",
       "\n",
       "                                host_strain_nomenclature engraftment_site  \\\n",
       "0                     NOD.CB-17-Prkdc <sup>scid</sup>/Rj            flank   \n",
       "1                     NOD.CB-17-Prkdc <sup>scid</sup>/Rj            flank   \n",
       "2                     NOD.CB-17-Prkdc <sup>scid</sup>/Rj            flank   \n",
       "3                     NOD.CB-17-Prkdc <sup>scid</sup>/Rj            flank   \n",
       "4                     NOD.CB-17-Prkdc <sup>scid</sup>/Rj            flank   \n",
       "...                                                  ...              ...   \n",
       "14083  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...     Subcutaneous   \n",
       "14084  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...     Subcutaneous   \n",
       "14085  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...     Subcutaneous   \n",
       "14086  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...     Subcutaneous   \n",
       "14087  NOD.Cg-Prkdc<sup>scid</sup> Il2rg<sup>tm1Wjl</...     Subcutaneous   \n",
       "\n",
       "      engraftment_type      sample_type   sample_state       passage_number  \\\n",
       "0          heterotopic  tissue fragment  cryopreserved          0,1,2,3,4,5   \n",
       "1          heterotopic  tissue fragment  cryopreserved                1,2,3   \n",
       "2          heterotopic  tissue fragment  cryopreserved      0,1,2,3,4,5,6,7   \n",
       "3          heterotopic  tissue fragment  cryopreserved            0,1,2,3,4   \n",
       "4          heterotopic  tissue fragment  cryopreserved  0,1,2,3,4,5,6,7,8,9   \n",
       "...                ...              ...            ...                  ...   \n",
       "14083      heterotopic  tissue fragment         frozen                   20   \n",
       "14084      heterotopic  tissue fragment         frozen                   20   \n",
       "14085      heterotopic  tissue fragment         frozen                   10   \n",
       "14086      heterotopic  tissue fragment         frozen                   10   \n",
       "14087      heterotopic  tissue fragment         frozen                   10   \n",
       "\n",
       "      parent_id origin_patient_sample_id  ... Unnamed: 1015 Unnamed: 1016  \\\n",
       "0           NaN                      NaN  ...           NaN           NaN   \n",
       "1           NaN                      NaN  ...           NaN           NaN   \n",
       "2           NaN                      NaN  ...           NaN           NaN   \n",
       "3           NaN                      NaN  ...           NaN           NaN   \n",
       "4           NaN                      NaN  ...           NaN           NaN   \n",
       "...         ...                      ...  ...           ...           ...   \n",
       "14083       NaN                      NaN  ...           NaN           NaN   \n",
       "14084       NaN                      NaN  ...           NaN           NaN   \n",
       "14085       NaN                      NaN  ...           NaN           NaN   \n",
       "14086       NaN                      NaN  ...           NaN           NaN   \n",
       "14087       NaN                      NaN  ...           NaN           NaN   \n",
       "\n",
       "      Unnamed: 1017 Unnamed: 1018 Unnamed: 1019 Unnamed: 1020 Unnamed: 1021  \\\n",
       "0               NaN           NaN           NaN           NaN           NaN   \n",
       "1               NaN           NaN           NaN           NaN           NaN   \n",
       "2               NaN           NaN           NaN           NaN           NaN   \n",
       "3               NaN           NaN           NaN           NaN           NaN   \n",
       "4               NaN           NaN           NaN           NaN           NaN   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "14083           NaN           NaN           NaN           NaN           NaN   \n",
       "14084           NaN           NaN           NaN           NaN           NaN   \n",
       "14085           NaN           NaN           NaN           NaN           NaN   \n",
       "14086           NaN           NaN           NaN           NaN           NaN   \n",
       "14087           NaN           NaN           NaN           NaN           NaN   \n",
       "\n",
       "      Unnamed: 1022 Unnamed: 1023 Unnamed: 1024  \n",
       "0               NaN           NaN           NaN  \n",
       "1               NaN           NaN           NaN  \n",
       "2               NaN           NaN           NaN  \n",
       "3               NaN           NaN           NaN  \n",
       "4               NaN           NaN           NaN  \n",
       "...             ...           ...           ...  \n",
       "14083           NaN           NaN           NaN  \n",
       "14084           NaN           NaN           NaN  \n",
       "14085           NaN           NaN           NaN  \n",
       "14086           NaN           NaN           NaN  \n",
       "14087           NaN           NaN           NaN  \n",
       "\n",
       "[14088 rows x 1071 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model_id</th>\n",
       "      <th>host_strain_name</th>\n",
       "      <th>host_strain_nomenclature</th>\n",
       "      <th>engraftment_site</th>\n",
       "      <th>engraftment_type</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sample_state</th>\n",
       "      <th>passage_number</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>origin_patient_sample_id</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 1015</th>\n",
       "      <th>Unnamed: 1016</th>\n",
       "      <th>Unnamed: 1017</th>\n",
       "      <th>Unnamed: 1018</th>\n",
       "      <th>Unnamed: 1019</th>\n",
       "      <th>Unnamed: 1020</th>\n",
       "      <th>Unnamed: 1021</th>\n",
       "      <th>Unnamed: 1022</th>\n",
       "      <th>Unnamed: 1023</th>\n",
       "      <th>Unnamed: 1024</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VHIO_PC-PDX-1</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.CB-17-Prkdc &lt;sup&gt;scid&lt;/sup&gt;/Rj</td>\n",
       "      <td>flank</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>cryopreserved</td>\n",
       "      <td>0,1,2,3,4,5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VHIO_PC-PDX-3</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.CB-17-Prkdc &lt;sup&gt;scid&lt;/sup&gt;/Rj</td>\n",
       "      <td>flank</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>cryopreserved</td>\n",
       "      <td>1,2,3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VHIO_PC-PDX-6</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.CB-17-Prkdc &lt;sup&gt;scid&lt;/sup&gt;/Rj</td>\n",
       "      <td>flank</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>cryopreserved</td>\n",
       "      <td>0,1,2,3,4,5,6,7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VHIO_PC-PDX-9</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.CB-17-Prkdc &lt;sup&gt;scid&lt;/sup&gt;/Rj</td>\n",
       "      <td>flank</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>cryopreserved</td>\n",
       "      <td>0,1,2,3,4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VHIO_PC-PDX-10</td>\n",
       "      <td>NOD SCID</td>\n",
       "      <td>NOD.CB-17-Prkdc &lt;sup&gt;scid&lt;/sup&gt;/Rj</td>\n",
       "      <td>flank</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>cryopreserved</td>\n",
       "      <td>0,1,2,3,4,5,6,7,8,9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14083</th>\n",
       "      <td>472M</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>Subcutaneous</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>frozen</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14084</th>\n",
       "      <td>472M</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>Subcutaneous</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>frozen</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14085</th>\n",
       "      <td>472M_Cx</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>Subcutaneous</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>frozen</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14086</th>\n",
       "      <td>472M_Cx</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>Subcutaneous</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>frozen</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14087</th>\n",
       "      <td>472M_Cx</td>\n",
       "      <td>NSG</td>\n",
       "      <td>NOD.Cg-Prkdc&lt;sup&gt;scid&lt;/sup&gt; Il2rg&lt;sup&gt;tm1Wjl&lt;/...</td>\n",
       "      <td>Subcutaneous</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>frozen</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14088 rows × 1071 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 151
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T08:09:18.953062Z",
     "start_time": "2025-04-22T08:09:18.873752Z"
    }
   },
   "cell_type": "code",
   "source": [
    "new_df = metadata[[\n",
    " 'host_strain_name',\n",
    " 'host_strain_nomenclature',\n",
    " 'engraftment_site',\n",
    " 'engraftment_type',\n",
    " 'sample_type',\n",
    " 'sample_state',\n",
    " 'passage_number',\n",
    " 'publications',\n",
    " 'supplier',\n",
    " 'supplier_type',\n",
    " 'catalog_number',\n",
    " 'vendor_link',\n",
    " 'external_ids',\n",
    " 'validation_technique',\n",
    " 'description',\n",
    " 'passages_tested',\n",
    " 'validation_host_strain_nomenclature',\n",
    " 'morphological_features',\n",
    " 'SNP_analysis',\n",
    " 'STR_analysis',\n",
    " 'tumour_status',\n",
    " 'model_purity',\n",
    " 'comments',\n",
    " 'collection_date',\n",
    " 'collection_event',\n",
    " 'months_since_collection_1',\n",
    " 'age_in_years_at_collection',\n",
    " 'diagnosis',\n",
    " 'tumour_type',\n",
    " 'primary_site',\n",
    " 'collection_site',\n",
    " 'collection_method',\n",
    " 'stage',\n",
    " 'staging_system',\n",
    " 'grade',\n",
    " 'grading_system',\n",
    " 'virology_status',\n",
    " 'gene_mutation_status',\n",
    " 'sharable',\n",
    " 'treatment_naive_at_collection',\n",
    " 'treated_at_collection',\n",
    " 'treated_prior_to_collection',\n",
    " 'response_to_treatment',\n",
    " 'sex',\n",
    " 'history',\n",
    " 'ethnicity',\n",
    " 'ethnicity_assessment_method',\n",
    " 'initial_diagnosis',\n",
    " 'age_at_initial_diagnosis',\n",
    " 'age_category',\n",
    " 'smoking_status',\n",
    " 'alcohol_status',\n",
    " 'alcohol_frequency',\n",
    " 'family_history_of_cancer']]"
   ],
   "id": "aba6d111b683518d",
   "outputs": [],
   "execution_count": 158
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T08:29:18.704962Z",
     "start_time": "2025-04-22T08:29:18.184811Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "\n",
    "\n",
    "# Get unique values for each column\n",
    "unique_cols = {col: new_df[col].str.lower().dropna().unique() for col in new_df.columns}\n",
    "\n",
    "# Find the max number of unique values among all columns\n",
    "max_len = max(len(v) for v in unique_cols.values())\n",
    "\n",
    "# Pad each column's unique values with NaN to align them vertically\n",
    "aligned_unique_df = pd.DataFrame({\n",
    "    col: pd.Series(vals).reindex(range(max_len))\n",
    "    for col, vals in unique_cols.items()\n",
    "})\n",
    "\n",
    "print(aligned_unique_df)\n"
   ],
   "id": "9ae2ad149d2ab422",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     host_strain_name                           host_strain_nomenclature  \\\n",
      "0            nod scid                 nod.cb-17-prkdc <sup>scid</sup>/rj   \n",
      "1                 nsg  nod.cg-prkdc<sup>scid</sup> il2rg<sup>tm1wjl</...   \n",
      "2          swiss nude                    nod.cb17-prkdc<sup>scid</sup>/j   \n",
      "3                 nog                  nod.cb17-prkdc<sup>scid</sup>/szj   \n",
      "4           nmri nude                     crl:nu(ico)-foxn1<sup>nu</sup>   \n",
      "...               ...                                                ...   \n",
      "1554              NaN                                                NaN   \n",
      "1555              NaN                                                NaN   \n",
      "1556              NaN                                                NaN   \n",
      "1557              NaN                                                NaN   \n",
      "1558              NaN                                                NaN   \n",
      "\n",
      "                           engraftment_site engraftment_type  \\\n",
      "0                                     flank      heterotopic   \n",
      "1                          peripheral blood       orthotopic   \n",
      "2                           mammary fat pad     not provided   \n",
      "3                    bone marrow and spleen     subcutaneous   \n",
      "4     intravenous, either lateral tail vein      intravenous   \n",
      "...                                     ...              ...   \n",
      "1554                                    NaN              NaN   \n",
      "1555                                    NaN              NaN   \n",
      "1556                                    NaN              NaN   \n",
      "1557                                    NaN              NaN   \n",
      "1558                                    NaN              NaN   \n",
      "\n",
      "                                 sample_type              sample_state  \\\n",
      "0                            tissue fragment             cryopreserved   \n",
      "1                            cell suspension  ffpe tissue mouse spleen   \n",
      "2                                bone marrow                      ffpe   \n",
      "3     bone marrow aspirates,peripheral blood                     fresh   \n",
      "4                                bone biospy                    frozen   \n",
      "...                                      ...                       ...   \n",
      "1554                                     NaN                       NaN   \n",
      "1555                                     NaN                       NaN   \n",
      "1556                                     NaN                       NaN   \n",
      "1557                                     NaN                       NaN   \n",
      "1558                                     NaN                       NaN   \n",
      "\n",
      "           passage_number    publications      supplier supplier_type  ...  \\\n",
      "0             0,1,2,3,4,5  pmid: 28581516  not provided  not provided  ...   \n",
      "1                   1,2,3  pmid: 29074954           hsm      academic  ...   \n",
      "2         0,1,2,3,4,5,6,7  pmid: 35819446           css      academia  ...   \n",
      "3               0,1,2,3,4  pmid: 27815673       hci-bcm           NaN  ...   \n",
      "4     0,1,2,3,4,5,6,7,8,9  pmid: 31693904           nki           NaN  ...   \n",
      "...                   ...             ...           ...           ...  ...   \n",
      "1554                  NaN             NaN           NaN           NaN  ...   \n",
      "1555                  NaN             NaN           NaN           NaN  ...   \n",
      "1556                  NaN             NaN           NaN           NaN  ...   \n",
      "1557                  NaN             NaN           NaN           NaN  ...   \n",
      "1558                  NaN             NaN           NaN           NaN  ...   \n",
      "\n",
      "                                                history         ethnicity  \\\n",
      "0                                          not provided      not provided   \n",
      "1     hereditary idiopathic hemochromatosis, alcohol...           italian   \n",
      "2     arterial hypertension, type 2 diabetes, hyperl...  western european   \n",
      "3     arterial hypertension, type 2 diabetes, obesit...  mixed or unknown   \n",
      "4     arterial hypertension, type 2 diabetes, obesit...          european   \n",
      "...                                                 ...               ...   \n",
      "1554                                                NaN               NaN   \n",
      "1555                                                NaN               NaN   \n",
      "1556                                                NaN               NaN   \n",
      "1557                                                NaN               NaN   \n",
      "1558                                                NaN               NaN   \n",
      "\n",
      "     ethnicity_assessment_method  \\\n",
      "0                   not provided   \n",
      "1                  self-assessed   \n",
      "2                        genetic   \n",
      "3                  not collected   \n",
      "4                            NaN   \n",
      "...                          ...   \n",
      "1554                         NaN   \n",
      "1555                         NaN   \n",
      "1556                         NaN   \n",
      "1557                         NaN   \n",
      "1558                         NaN   \n",
      "\n",
      "                                      initial_diagnosis  \\\n",
      "0     pancreatic carcinoma - infiltrating ductal ade...   \n",
      "1     pancreatic carcinoma - infiltrating ductal ade...   \n",
      "2     pancreatic carcinoma - intraductal papillary m...   \n",
      "3                                 pancreatic carcinoma    \n",
      "4                                    cholangiocarcinoma   \n",
      "...                                                 ...   \n",
      "1554                                                NaN   \n",
      "1555                                                NaN   \n",
      "1556                                                NaN   \n",
      "1557                                                NaN   \n",
      "1558                                                NaN   \n",
      "\n",
      "     age_at_initial_diagnosis  age_category  smoking_status alcohol_status  \\\n",
      "0                not provided  not provided    not provided   not provided   \n",
      "1                          49         adult       ex-smoker            yes   \n",
      "2                          70     pediatric      non-smoker             no   \n",
      "3                          55           NaN  current smoker  not collected   \n",
      "4                          54           NaN      non smoker            NaN   \n",
      "...                       ...           ...             ...            ...   \n",
      "1554                      NaN           NaN             NaN            NaN   \n",
      "1555                      NaN           NaN             NaN            NaN   \n",
      "1556                      NaN           NaN             NaN            NaN   \n",
      "1557                      NaN           NaN             NaN            NaN   \n",
      "1558                      NaN           NaN             NaN            NaN   \n",
      "\n",
      "     alcohol_frequency                           family_history_of_cancer  \n",
      "0         not provided                                       not provided  \n",
      "1        not collected                                                yes  \n",
      "2                  NaN                                                 no  \n",
      "3                  NaN   family history: 1st degree relative, lung cancer  \n",
      "4                  NaN  family history: 1st degree relative, lung ca; ...  \n",
      "...                ...                                                ...  \n",
      "1554               NaN                                                NaN  \n",
      "1555               NaN                                                NaN  \n",
      "1556               NaN                                                NaN  \n",
      "1557               NaN                                                NaN  \n",
      "1558               NaN                                                NaN  \n",
      "\n",
      "[1559 rows x 54 columns]\n"
     ]
    }
   ],
   "execution_count": 166
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T08:29:52.466505Z",
     "start_time": "2025-04-22T08:29:52.365697Z"
    }
   },
   "cell_type": "code",
   "source": "aligned_unique_df[[ 'sex', 'age_in_years_at_collection', 'diagnosis', 'tumour_type', 'primary_site', 'collection_site', 'sharable', 'treatment_naive_at_collection', 'host_strain_name', 'host_strain_nomenclature', 'engraftment_site', 'engraftment_type', 'sample_type', 'passage_number', 'supplier', 'supplier_type', 'catalog_number', 'vendor_link', 'validation_technique', 'description', 'passages_tested', 'validation_host_strain_nomenclature', 'SNP_analysis', 'STR_analysis', 'comments']].fillna('')",
   "id": "c438574335f0a0f8",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                sex age_in_years_at_collection  \\\n",
       "0            female               not provided   \n",
       "1              male                         69   \n",
       "2      not provided                         74   \n",
       "3     not collected                         59   \n",
       "4             other                         54   \n",
       "...             ...                        ...   \n",
       "1554                                             \n",
       "1555                                             \n",
       "1556                                             \n",
       "1557                                             \n",
       "1558                                             \n",
       "\n",
       "                                              diagnosis    tumour_type  \\\n",
       "0     pancreatic carcinoma - infiltrating ductal ade...        primary   \n",
       "1     pancreatic carcinoma - infiltrating ductal ade...     metastatic   \n",
       "2     pancreatic carcinoma - intraductal papillary m...   not provided   \n",
       "3                                  pancreatic carcinoma      recurrent   \n",
       "4                                    cholangiocarcinoma  not collected   \n",
       "...                                                 ...            ...   \n",
       "1554                                                                     \n",
       "1555                                                                     \n",
       "1556                                                                     \n",
       "1557                                                                     \n",
       "1558                                                                     \n",
       "\n",
       "            primary_site collection_site      sharable  \\\n",
       "0               pancreas        pancreas  not provided   \n",
       "1              bile duct           liver           yes   \n",
       "2       peripheral blood      peritoneum            no   \n",
       "3                 breast            lung                 \n",
       "4     spleen/bone marrow    not provided                 \n",
       "...                  ...             ...           ...   \n",
       "1554                                                     \n",
       "1555                                                     \n",
       "1556                                                     \n",
       "1557                                                     \n",
       "1558                                                     \n",
       "\n",
       "     treatment_naive_at_collection host_strain_name  \\\n",
       "0                     not provided         nod scid   \n",
       "1                              yes              nsg   \n",
       "2                               no       swiss nude   \n",
       "3                    not collected              nog   \n",
       "4                              no         nmri nude   \n",
       "...                            ...              ...   \n",
       "1554                                                  \n",
       "1555                                                  \n",
       "1556                                                  \n",
       "1557                                                  \n",
       "1558                                                  \n",
       "\n",
       "                               host_strain_nomenclature  ... supplier_type  \\\n",
       "0                    nod.cb-17-prkdc <sup>scid</sup>/rj  ...  not provided   \n",
       "1     nod.cg-prkdc<sup>scid</sup> il2rg<sup>tm1wjl</...  ...      academic   \n",
       "2                       nod.cb17-prkdc<sup>scid</sup>/j  ...      academia   \n",
       "3                     nod.cb17-prkdc<sup>scid</sup>/szj  ...                 \n",
       "4                        crl:nu(ico)-foxn1<sup>nu</sup>  ...                 \n",
       "...                                                 ...  ...           ...   \n",
       "1554                                                     ...                 \n",
       "1555                                                     ...                 \n",
       "1556                                                     ...                 \n",
       "1557                                                     ...                 \n",
       "1558                                                     ...                 \n",
       "\n",
       "              catalog_number                       vendor_link  \\\n",
       "0               not provided                      not provided   \n",
       "1               gc0015_nsg_2  https://www.cccells.org/rclp.php   \n",
       "2               cm18_nsg_345                                     \n",
       "3                pm129_nsg_6                                     \n",
       "4     breast_hsm_bc004.1_pdx                                     \n",
       "...                      ...                               ...   \n",
       "1554                 463.45a                                     \n",
       "1555                 463.55a                                     \n",
       "1556                 463.58a                                     \n",
       "1557                    472m                                     \n",
       "1558                 472m cx                                     \n",
       "\n",
       "                                   validation_technique  \\\n",
       "0                                  immunohistochemistry   \n",
       "1     immunohystochemistry of mouse spleen, flow cyt...   \n",
       "2     immunohystochemistry of mouse spleen, flow cyt...   \n",
       "3       immunohystochemistry of human bc grown in mouse   \n",
       "4                                          not provided   \n",
       "...                                                 ...   \n",
       "1554                                                      \n",
       "1555                                                      \n",
       "1556                                                      \n",
       "1557                                                      \n",
       "1558                                                      \n",
       "\n",
       "                                            description passages_tested  \\\n",
       "0                                          not provided    not provided   \n",
       "1     good concordance between xenograft and patient...               0   \n",
       "2     concordance between xenograft and patient tumo...               1   \n",
       "3     some derivation  between xenograft and  patien...               6   \n",
       "4     high concordance between xenograft and  patien...               3   \n",
       "...                                                 ...             ...   \n",
       "1554                                                                      \n",
       "1555                                                                      \n",
       "1556                                                                      \n",
       "1557                                                                      \n",
       "1558                                                                      \n",
       "\n",
       "                    validation_host_strain_nomenclature  SNP_analysis  \\\n",
       "0                    nod.cb-17-prkdc <sup>scid</sup>/rj  not provided   \n",
       "1     nod.cg-prkdc<sup>scid</sup> il2rg<sup>tm1wjl</...            no   \n",
       "2                       nod.cb17-prkdc<sup>scid</sup>/j                 \n",
       "3                     nod.cb17-prkdc<sup>scid</sup>/szj                 \n",
       "4                        crl:nu(ico)-foxn1<sup>nu</sup>                 \n",
       "...                                                 ...           ...   \n",
       "1554                                                                    \n",
       "1555                                                                    \n",
       "1556                                                                    \n",
       "1557                                                                    \n",
       "1558                                                                    \n",
       "\n",
       "      STR_analysis                                           comments  \n",
       "0     not provided                                       not provided  \n",
       "1               no  the childhood cancer repository has a matching...  \n",
       "2              yes  the childhood cancer repository has a matching...  \n",
       "3                   the childhood cancer repository has a matching...  \n",
       "4                   the childhood cancer repository has a matching...  \n",
       "...            ...                                                ...  \n",
       "1554                                                                   \n",
       "1555                                                                   \n",
       "1556                                                                   \n",
       "1557                                                                   \n",
       "1558                                                                   \n",
       "\n",
       "[1559 rows x 25 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sex</th>\n",
       "      <th>age_in_years_at_collection</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>tumour_type</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>collection_site</th>\n",
       "      <th>sharable</th>\n",
       "      <th>treatment_naive_at_collection</th>\n",
       "      <th>host_strain_name</th>\n",
       "      <th>host_strain_nomenclature</th>\n",
       "      <th>...</th>\n",
       "      <th>supplier_type</th>\n",
       "      <th>catalog_number</th>\n",
       "      <th>vendor_link</th>\n",
       "      <th>validation_technique</th>\n",
       "      <th>description</th>\n",
       "      <th>passages_tested</th>\n",
       "      <th>validation_host_strain_nomenclature</th>\n",
       "      <th>SNP_analysis</th>\n",
       "      <th>STR_analysis</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>female</td>\n",
       "      <td>not provided</td>\n",
       "      <td>pancreatic carcinoma - infiltrating ductal ade...</td>\n",
       "      <td>primary</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>nod scid</td>\n",
       "      <td>nod.cb-17-prkdc &lt;sup&gt;scid&lt;/sup&gt;/rj</td>\n",
       "      <td>...</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>immunohistochemistry</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>nod.cb-17-prkdc &lt;sup&gt;scid&lt;/sup&gt;/rj</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>male</td>\n",
       "      <td>69</td>\n",
       "      <td>pancreatic carcinoma - infiltrating ductal ade...</td>\n",
       "      <td>metastatic</td>\n",
       "      <td>bile duct</td>\n",
       "      <td>liver</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>nsg</td>\n",
       "      <td>nod.cg-prkdc&lt;sup&gt;scid&lt;/sup&gt; il2rg&lt;sup&gt;tm1wjl&lt;/...</td>\n",
       "      <td>...</td>\n",
       "      <td>academic</td>\n",
       "      <td>gc0015_nsg_2</td>\n",
       "      <td>https://www.cccells.org/rclp.php</td>\n",
       "      <td>immunohystochemistry of mouse spleen, flow cyt...</td>\n",
       "      <td>good concordance between xenograft and patient...</td>\n",
       "      <td>0</td>\n",
       "      <td>nod.cg-prkdc&lt;sup&gt;scid&lt;/sup&gt; il2rg&lt;sup&gt;tm1wjl&lt;/...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>the childhood cancer repository has a matching...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>not provided</td>\n",
       "      <td>74</td>\n",
       "      <td>pancreatic carcinoma - intraductal papillary m...</td>\n",
       "      <td>not provided</td>\n",
       "      <td>peripheral blood</td>\n",
       "      <td>peritoneum</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>swiss nude</td>\n",
       "      <td>nod.cb17-prkdc&lt;sup&gt;scid&lt;/sup&gt;/j</td>\n",
       "      <td>...</td>\n",
       "      <td>academia</td>\n",
       "      <td>cm18_nsg_345</td>\n",
       "      <td></td>\n",
       "      <td>immunohystochemistry of mouse spleen, flow cyt...</td>\n",
       "      <td>concordance between xenograft and patient tumo...</td>\n",
       "      <td>1</td>\n",
       "      <td>nod.cb17-prkdc&lt;sup&gt;scid&lt;/sup&gt;/j</td>\n",
       "      <td></td>\n",
       "      <td>yes</td>\n",
       "      <td>the childhood cancer repository has a matching...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>not collected</td>\n",
       "      <td>59</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>recurrent</td>\n",
       "      <td>breast</td>\n",
       "      <td>lung</td>\n",
       "      <td></td>\n",
       "      <td>not collected</td>\n",
       "      <td>nog</td>\n",
       "      <td>nod.cb17-prkdc&lt;sup&gt;scid&lt;/sup&gt;/szj</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>pm129_nsg_6</td>\n",
       "      <td></td>\n",
       "      <td>immunohystochemistry of human bc grown in mouse</td>\n",
       "      <td>some derivation  between xenograft and  patien...</td>\n",
       "      <td>6</td>\n",
       "      <td>nod.cb17-prkdc&lt;sup&gt;scid&lt;/sup&gt;/szj</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>the childhood cancer repository has a matching...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>other</td>\n",
       "      <td>54</td>\n",
       "      <td>cholangiocarcinoma</td>\n",
       "      <td>not collected</td>\n",
       "      <td>spleen/bone marrow</td>\n",
       "      <td>not provided</td>\n",
       "      <td></td>\n",
       "      <td>no</td>\n",
       "      <td>nmri nude</td>\n",
       "      <td>crl:nu(ico)-foxn1&lt;sup&gt;nu&lt;/sup&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>breast_hsm_bc004.1_pdx</td>\n",
       "      <td></td>\n",
       "      <td>not provided</td>\n",
       "      <td>high concordance between xenograft and  patien...</td>\n",
       "      <td>3</td>\n",
       "      <td>crl:nu(ico)-foxn1&lt;sup&gt;nu&lt;/sup&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>the childhood cancer repository has a matching...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>463.45a</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1555</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>463.55a</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>463.58a</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>472m</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1558</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>472m cx</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1559 rows × 25 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 168
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T08:35:14.768596Z",
     "start_time": "2025-04-22T08:35:14.659740Z"
    }
   },
   "cell_type": "code",
   "source": "aligned_unique_df.fillna('')",
   "id": "383b79b8c058f612",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     host_strain_name                           host_strain_nomenclature  \\\n",
       "0            nod scid                 nod.cb-17-prkdc <sup>scid</sup>/rj   \n",
       "1                 nsg  nod.cg-prkdc<sup>scid</sup> il2rg<sup>tm1wjl</...   \n",
       "2          swiss nude                    nod.cb17-prkdc<sup>scid</sup>/j   \n",
       "3                 nog                  nod.cb17-prkdc<sup>scid</sup>/szj   \n",
       "4           nmri nude                     crl:nu(ico)-foxn1<sup>nu</sup>   \n",
       "...               ...                                                ...   \n",
       "1554                                                                       \n",
       "1555                                                                       \n",
       "1556                                                                       \n",
       "1557                                                                       \n",
       "1558                                                                       \n",
       "\n",
       "                           engraftment_site engraftment_type  \\\n",
       "0                                     flank      heterotopic   \n",
       "1                          peripheral blood       orthotopic   \n",
       "2                           mammary fat pad     not provided   \n",
       "3                    bone marrow and spleen     subcutaneous   \n",
       "4     intravenous, either lateral tail vein      intravenous   \n",
       "...                                     ...              ...   \n",
       "1554                                                           \n",
       "1555                                                           \n",
       "1556                                                           \n",
       "1557                                                           \n",
       "1558                                                           \n",
       "\n",
       "                                 sample_type              sample_state  \\\n",
       "0                            tissue fragment             cryopreserved   \n",
       "1                            cell suspension  ffpe tissue mouse spleen   \n",
       "2                                bone marrow                      ffpe   \n",
       "3     bone marrow aspirates,peripheral blood                     fresh   \n",
       "4                                bone biospy                    frozen   \n",
       "...                                      ...                       ...   \n",
       "1554                                                                     \n",
       "1555                                                                     \n",
       "1556                                                                     \n",
       "1557                                                                     \n",
       "1558                                                                     \n",
       "\n",
       "           passage_number    publications      supplier supplier_type  ...  \\\n",
       "0             0,1,2,3,4,5  pmid: 28581516  not provided  not provided  ...   \n",
       "1                   1,2,3  pmid: 29074954           hsm      academic  ...   \n",
       "2         0,1,2,3,4,5,6,7  pmid: 35819446           css      academia  ...   \n",
       "3               0,1,2,3,4  pmid: 27815673       hci-bcm                ...   \n",
       "4     0,1,2,3,4,5,6,7,8,9  pmid: 31693904           nki                ...   \n",
       "...                   ...             ...           ...           ...  ...   \n",
       "1554                                                                   ...   \n",
       "1555                                                                   ...   \n",
       "1556                                                                   ...   \n",
       "1557                                                                   ...   \n",
       "1558                                                                   ...   \n",
       "\n",
       "                                                history         ethnicity  \\\n",
       "0                                          not provided      not provided   \n",
       "1     hereditary idiopathic hemochromatosis, alcohol...           italian   \n",
       "2     arterial hypertension, type 2 diabetes, hyperl...  western european   \n",
       "3     arterial hypertension, type 2 diabetes, obesit...  mixed or unknown   \n",
       "4     arterial hypertension, type 2 diabetes, obesit...          european   \n",
       "...                                                 ...               ...   \n",
       "1554                                                                        \n",
       "1555                                                                        \n",
       "1556                                                                        \n",
       "1557                                                                        \n",
       "1558                                                                        \n",
       "\n",
       "     ethnicity_assessment_method  \\\n",
       "0                   not provided   \n",
       "1                  self-assessed   \n",
       "2                        genetic   \n",
       "3                  not collected   \n",
       "4                                  \n",
       "...                          ...   \n",
       "1554                               \n",
       "1555                               \n",
       "1556                               \n",
       "1557                               \n",
       "1558                               \n",
       "\n",
       "                                      initial_diagnosis  \\\n",
       "0     pancreatic carcinoma - infiltrating ductal ade...   \n",
       "1     pancreatic carcinoma - infiltrating ductal ade...   \n",
       "2     pancreatic carcinoma - intraductal papillary m...   \n",
       "3                                 pancreatic carcinoma    \n",
       "4                                    cholangiocarcinoma   \n",
       "...                                                 ...   \n",
       "1554                                                      \n",
       "1555                                                      \n",
       "1556                                                      \n",
       "1557                                                      \n",
       "1558                                                      \n",
       "\n",
       "     age_at_initial_diagnosis  age_category  smoking_status alcohol_status  \\\n",
       "0                not provided  not provided    not provided   not provided   \n",
       "1                          49         adult       ex-smoker            yes   \n",
       "2                          70     pediatric      non-smoker             no   \n",
       "3                          55                current smoker  not collected   \n",
       "4                          54                    non smoker                  \n",
       "...                       ...           ...             ...            ...   \n",
       "1554                                                                         \n",
       "1555                                                                         \n",
       "1556                                                                         \n",
       "1557                                                                         \n",
       "1558                                                                         \n",
       "\n",
       "     alcohol_frequency                           family_history_of_cancer  \n",
       "0         not provided                                       not provided  \n",
       "1        not collected                                                yes  \n",
       "2                                                                      no  \n",
       "3                        family history: 1st degree relative, lung cancer  \n",
       "4                       family history: 1st degree relative, lung ca; ...  \n",
       "...                ...                                                ...  \n",
       "1554                                                                       \n",
       "1555                                                                       \n",
       "1556                                                                       \n",
       "1557                                                                       \n",
       "1558                                                                       \n",
       "\n",
       "[1559 rows x 54 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>host_strain_name</th>\n",
       "      <th>host_strain_nomenclature</th>\n",
       "      <th>engraftment_site</th>\n",
       "      <th>engraftment_type</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sample_state</th>\n",
       "      <th>passage_number</th>\n",
       "      <th>publications</th>\n",
       "      <th>supplier</th>\n",
       "      <th>supplier_type</th>\n",
       "      <th>...</th>\n",
       "      <th>history</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>ethnicity_assessment_method</th>\n",
       "      <th>initial_diagnosis</th>\n",
       "      <th>age_at_initial_diagnosis</th>\n",
       "      <th>age_category</th>\n",
       "      <th>smoking_status</th>\n",
       "      <th>alcohol_status</th>\n",
       "      <th>alcohol_frequency</th>\n",
       "      <th>family_history_of_cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>nod scid</td>\n",
       "      <td>nod.cb-17-prkdc &lt;sup&gt;scid&lt;/sup&gt;/rj</td>\n",
       "      <td>flank</td>\n",
       "      <td>heterotopic</td>\n",
       "      <td>tissue fragment</td>\n",
       "      <td>cryopreserved</td>\n",
       "      <td>0,1,2,3,4,5</td>\n",
       "      <td>pmid: 28581516</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>...</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>pancreatic carcinoma - infiltrating ductal ade...</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>nsg</td>\n",
       "      <td>nod.cg-prkdc&lt;sup&gt;scid&lt;/sup&gt; il2rg&lt;sup&gt;tm1wjl&lt;/...</td>\n",
       "      <td>peripheral blood</td>\n",
       "      <td>orthotopic</td>\n",
       "      <td>cell suspension</td>\n",
       "      <td>ffpe tissue mouse spleen</td>\n",
       "      <td>1,2,3</td>\n",
       "      <td>pmid: 29074954</td>\n",
       "      <td>hsm</td>\n",
       "      <td>academic</td>\n",
       "      <td>...</td>\n",
       "      <td>hereditary idiopathic hemochromatosis, alcohol...</td>\n",
       "      <td>italian</td>\n",
       "      <td>self-assessed</td>\n",
       "      <td>pancreatic carcinoma - infiltrating ductal ade...</td>\n",
       "      <td>49</td>\n",
       "      <td>adult</td>\n",
       "      <td>ex-smoker</td>\n",
       "      <td>yes</td>\n",
       "      <td>not collected</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>swiss nude</td>\n",
       "      <td>nod.cb17-prkdc&lt;sup&gt;scid&lt;/sup&gt;/j</td>\n",
       "      <td>mammary fat pad</td>\n",
       "      <td>not provided</td>\n",
       "      <td>bone marrow</td>\n",
       "      <td>ffpe</td>\n",
       "      <td>0,1,2,3,4,5,6,7</td>\n",
       "      <td>pmid: 35819446</td>\n",
       "      <td>css</td>\n",
       "      <td>academia</td>\n",
       "      <td>...</td>\n",
       "      <td>arterial hypertension, type 2 diabetes, hyperl...</td>\n",
       "      <td>western european</td>\n",
       "      <td>genetic</td>\n",
       "      <td>pancreatic carcinoma - intraductal papillary m...</td>\n",
       "      <td>70</td>\n",
       "      <td>pediatric</td>\n",
       "      <td>non-smoker</td>\n",
       "      <td>no</td>\n",
       "      <td></td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nog</td>\n",
       "      <td>nod.cb17-prkdc&lt;sup&gt;scid&lt;/sup&gt;/szj</td>\n",
       "      <td>bone marrow and spleen</td>\n",
       "      <td>subcutaneous</td>\n",
       "      <td>bone marrow aspirates,peripheral blood</td>\n",
       "      <td>fresh</td>\n",
       "      <td>0,1,2,3,4</td>\n",
       "      <td>pmid: 27815673</td>\n",
       "      <td>hci-bcm</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>arterial hypertension, type 2 diabetes, obesit...</td>\n",
       "      <td>mixed or unknown</td>\n",
       "      <td>not collected</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>55</td>\n",
       "      <td></td>\n",
       "      <td>current smoker</td>\n",
       "      <td>not collected</td>\n",
       "      <td></td>\n",
       "      <td>family history: 1st degree relative, lung cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>nmri nude</td>\n",
       "      <td>crl:nu(ico)-foxn1&lt;sup&gt;nu&lt;/sup&gt;</td>\n",
       "      <td>intravenous, either lateral tail vein</td>\n",
       "      <td>intravenous</td>\n",
       "      <td>bone biospy</td>\n",
       "      <td>frozen</td>\n",
       "      <td>0,1,2,3,4,5,6,7,8,9</td>\n",
       "      <td>pmid: 31693904</td>\n",
       "      <td>nki</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>arterial hypertension, type 2 diabetes, obesit...</td>\n",
       "      <td>european</td>\n",
       "      <td></td>\n",
       "      <td>cholangiocarcinoma</td>\n",
       "      <td>54</td>\n",
       "      <td></td>\n",
       "      <td>non smoker</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>family history: 1st degree relative, lung ca; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1555</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1558</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1559 rows × 54 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 170
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-22T08:14:01.736616Z",
     "start_time": "2025-04-22T08:14:01.156639Z"
    }
   },
   "cell_type": "code",
   "source": "pd.read_json('https://www.cancermodels.org/api/search_index?data_source=eq.WUSTL&select=external_model_id,histology')",
   "id": "8bcbd80b31497cb2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "    external_model_id                    histology\n",
       "0          WU-0014-T1         Colon Adenocarcinoma\n",
       "1        WUSTL WHIM14  Basal-Like Breast Carcinoma\n",
       "2         WUSTL SHIM9             Synovial Sarcoma\n",
       "3        WUSTL 369-06    Colorectal Adenocarcinoma\n",
       "4        WUSTL WHIM30  Basal-Like Breast Carcinoma\n",
       "..                ...                          ...\n",
       "295      WUSTL 655-08    Colorectal Adenocarcinoma\n",
       "296           WU-0075         Urothelial Carcinoma\n",
       "297      WUSTL 268-06    Colorectal Adenocarcinoma\n",
       "298        WU-0255-T1          Lung Adenocarcinoma\n",
       "299        WU-0082-T1    Invasive Breast Carcinoma\n",
       "\n",
       "[300 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>external_model_id</th>\n",
       "      <th>histology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WU-0014-T1</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WUSTL WHIM14</td>\n",
       "      <td>Basal-Like Breast Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WUSTL SHIM9</td>\n",
       "      <td>Synovial Sarcoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WUSTL 369-06</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WUSTL WHIM30</td>\n",
       "      <td>Basal-Like Breast Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>WUSTL 655-08</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>WU-0075</td>\n",
       "      <td>Urothelial Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>WUSTL 268-06</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>WU-0255-T1</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>WU-0082-T1</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>300 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 161
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "2f4070cbfeda5b77"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
