{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2024-12-11T11:14:43.355584Z",
     "start_time": "2024-12-11T11:14:41.645497Z"
    }
   },
   "source": [
    "import pandas as pd\n",
    "from Bio import Entrez\n",
    "from pycparser.ply.yacc import token\n",
    "\n",
    "from utils import get_files, join\n",
    "from requests import post, exceptions, get\n",
    "import xml.etree.ElementTree as ET\n",
    "import os\n",
    "import re"
   ],
   "outputs": [],
   "execution_count": 1
  },
  {
   "cell_type": "code",
   "source": [
    "def replace_html_tags(text):\n",
    "    text = re.sub(r'<sub>(.*?)</sub>', r'_\\1_', text)  # Replace <sub> tags with underscores for subscript\n",
    "    return text\n",
    "\n",
    "def get_abstract_from_api(path):\n",
    "    papers = pd.read_csv(path, sep='\\t')\n",
    "    if 'pid' not in papers.columns:\n",
    "        papers['pid'] = papers['pubmed_ids'].str.replace(' ', '')\n",
    "    papers['pmid'] = papers['pid'].astype(str).str.replace('PMID:', '')\n",
    "    ids = ','.join(papers['pmid'].to_list())\n",
    "    payload = {'db':'pubmed', 'id':ids}\n",
    "    response = post(f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi\", data=payload)\n",
    "    #print(f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id={ids}\")\n",
    "    root = ET.fromstring(response.text)\n",
    "\n",
    "    # Extract information\n",
    "    articles = []\n",
    "    for article in root.findall('PubmedArticle'):\n",
    "        pmid = article.find('.//PMID').text\n",
    "        title = article.find('.//ArticleTitle').text\n",
    "        journal_title = article.find('.//Journal//Title').text\n",
    "        abstract = article.findall('.//AbstractText')\n",
    "        abstract_text = \"\"\n",
    "        for abs in abstract:\n",
    "            if abs.text is not None:\n",
    "                text = ET.tostring(abs, encoding='unicode', method='text')\n",
    "                abstract_text += replace_html_tags(text)\n",
    "        pub_date_element = article.find('.//JournalIssue/PubDate')\n",
    "        year = pub_date_element.find('Year').text if pub_date_element.find('Year') is not None else ''\n",
    "        month = pub_date_element.find('Month').text if pub_date_element.find('Month') is not None else ''\n",
    "        day = pub_date_element.find('Day').text if pub_date_element.find('Day') is not None else ''\n",
    "\n",
    "        # Format the publication date\n",
    "        publication_date = f\"{month}-{year}\"\n",
    "        articles.append({\n",
    "            'PMID': pmid,\n",
    "            'Title': title,\n",
    "            'Journal': journal_title,\n",
    "            'Abstract': abstract_text,\n",
    "            'Date': publication_date\n",
    "            \n",
    "        })\n",
    "    abstract_df = pd.DataFrame(articles)\n",
    "    abstract_df = papers.merge(abstract_df, left_on='pmid', right_on='PMID', how='left')\n",
    "    abstract_df.to_csv('corpus_negative_abtract.tsv', index=False, sep='\\t')\n",
    "    for pubmed_article in abstract_df.iterrows():\n",
    "        filename = f\"corpus/{pubmed_article[1]['pid'].replace(':', '_')}.txt\"\n",
    "        with open(filename, 'w') as file:\n",
    "            file.write(str(pubmed_article[1]['Abstract']))\n",
    "    return abstract_df\n",
    "\n",
    "def read_pmid_get_abstract(path):\n",
    "    papers = pd.read_csv(path, sep='\\t')\n",
    "    Entrez.email = 'tushar@ebi.ac.uk'\n",
    "    if 'pid' not in papers.columns:\n",
    "        papers['pid'] = papers['pubmed_ids'].str.replace(' ', '')\n",
    "    papers['pmid'] = papers['pid'].astype(str).str.replace('PMID:', '')\n",
    "    pmids = papers['pid'].str.replace('PMID:', '').to_list()\n",
    "    handle = Entrez.efetch(db=\"pubmed\", id=','.join(map(str, pmids)),\n",
    "                           rettype=\"xml\", retmode=\"text\")\n",
    "    records = Entrez.read(handle)\n",
    "    abstracts = [(pubmed_article['MedlineCitation']['PMID'].strip(), pubmed_article['MedlineCitation']['Article']['Abstract']['AbstractText']) if 'Abstract' in pubmed_article['MedlineCitation']['Article'].keys() else (pubmed_article['MedlineCitation']['PMID'].strip(), 'Not provided')\n",
    "                 for pubmed_article in records['PubmedArticle']]\n",
    "    abstract_dict = dict(abstracts)\n",
    "    abstract_df = pd.DataFrame([abstract_dict.keys(), abstract_dict.values()]).transpose()\n",
    "    abstract_df = abstract_df.rename({0: 'pmid', 1: 'abstract'},axis=1)\n",
    "    papers = papers.merge(abstract_df, on='pmid', how='outer')\n",
    "    papers.to_csv('abstract_pmids_api_full.tsv', index=False, sep='\\t')\n",
    "    for pubmed_article in papers.iterrows():\n",
    "        filename = f\"new_abstracts_full/{pubmed_article[1]['pid'].replace(':', '_')}.txt\"\n",
    "        with open(filename, 'w') as file:\n",
    "            file.write(str(pubmed_article[1]['abstract']))\n",
    "    return papers"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-07-19T13:00:34.966712Z",
     "start_time": "2024-07-19T13:00:34.942805Z"
    }
   },
   "id": "bcc44e9821d69b5a",
   "outputs": [],
   "execution_count": 24
  },
  {
   "cell_type": "code",
   "source": [
    "abstract_df = get_abstract_from_api(\"corpus_negative.tsv\")\n",
    "abstract_df"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-07-03T13:11:06.083135Z",
     "start_time": "2024-07-03T13:11:03.054960Z"
    }
   },
   "id": "1af864e20bae35bd",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         pubmed_ids            pid      pmid      PMID  \\\n",
       "0          29860986       29860986  29860986  29860986   \n",
       "1          27741350       27741350  27741350  27741350   \n",
       "2          28574057       28574057  28574057  28574057   \n",
       "3          33504580       33504580  33504580  33504580   \n",
       "4          27839715       27839715  27839715  27839715   \n",
       "..              ...            ...       ...       ...   \n",
       "126  PMID: 28232476  PMID:28232476  28232476  28232476   \n",
       "127  PMID: 27815673  PMID:27815673  27815673  27815673   \n",
       "128  PMID: 29079660  PMID:29079660  29079660  29079660   \n",
       "129   PMID:28473534  PMID:28473534  28473534  28473534   \n",
       "130  PMID: 29093017  PMID:29093017  29093017  29093017   \n",
       "\n",
       "                                                 Title  \\\n",
       "0    Immunotherapy and Prevention of Pancreatic Can...   \n",
       "1          Measuring cancer evolution from the genome.   \n",
       "2    New methods in the diagnosis of cancer and gen...   \n",
       "3    Metabolic Codependencies in the Tumor Microenv...   \n",
       "4         [Lucy's cancer(s): A prehistorical origin?].   \n",
       "..                                                 ...   \n",
       "126  Potential Targets' Analysis Reveals Dual PI3K/...   \n",
       "127  Erratum to: Patient-Derived Mammosphere and Xe...   \n",
       "128   Dual mTOR Kinase Inhibitor MLN0128 Sensitizes HR   \n",
       "129  Elacestrant (RAD1901), a Selective Estrogen Re...   \n",
       "130  A Population of Heterogeneous Breast Cancer Pa...   \n",
       "\n",
       "                                               Journal  \\\n",
       "0                                     Trends in cancer   \n",
       "1                             The Journal of pathology   \n",
       "2                                  Cancer gene therapy   \n",
       "3                                     Cancer discovery   \n",
       "4                 Gynecologie, obstetrique & fertilite   \n",
       "..                                                 ...   \n",
       "126  Clinical cancer research : an official journal...   \n",
       "127     Journal of mammary gland biology and neoplasia   \n",
       "128  Clinical cancer research : an official journal...   \n",
       "129  Clinical cancer research : an official journal...   \n",
       "130  Clinical cancer research : an official journal...   \n",
       "\n",
       "                                              Abstract      Date  \n",
       "0    Pancreatic cancer is the third-leading cause o...  Jun-2018  \n",
       "1    The temporal dynamics of cancer evolution rema...  Jan-2017  \n",
       "2    Cancer is one of the leading cause of death in...  Jun-2017  \n",
       "3    Metabolic reprogramming enables cancer cell gr...  May-2021  \n",
       "4    The recent discovery of the earliest hominin c...  Dec-2016  \n",
       "..                                                 ...       ...  \n",
       "126                                                     Mar-2017  \n",
       "127                                                     Dec-2016  \n",
       "128                                                     Jan-2018  \n",
       "129                                                     Aug-2017  \n",
       "130                                                     Nov-2017  \n",
       "\n",
       "[131 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_ids</th>\n",
       "      <th>pid</th>\n",
       "      <th>pmid</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Title</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Abstract</th>\n",
       "      <th>Date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29860986</td>\n",
       "      <td>29860986</td>\n",
       "      <td>29860986</td>\n",
       "      <td>29860986</td>\n",
       "      <td>Immunotherapy and Prevention of Pancreatic Can...</td>\n",
       "      <td>Trends in cancer</td>\n",
       "      <td>Pancreatic cancer is the third-leading cause o...</td>\n",
       "      <td>Jun-2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>27741350</td>\n",
       "      <td>27741350</td>\n",
       "      <td>27741350</td>\n",
       "      <td>27741350</td>\n",
       "      <td>Measuring cancer evolution from the genome.</td>\n",
       "      <td>The Journal of pathology</td>\n",
       "      <td>The temporal dynamics of cancer evolution rema...</td>\n",
       "      <td>Jan-2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>28574057</td>\n",
       "      <td>28574057</td>\n",
       "      <td>28574057</td>\n",
       "      <td>28574057</td>\n",
       "      <td>New methods in the diagnosis of cancer and gen...</td>\n",
       "      <td>Cancer gene therapy</td>\n",
       "      <td>Cancer is one of the leading cause of death in...</td>\n",
       "      <td>Jun-2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>33504580</td>\n",
       "      <td>33504580</td>\n",
       "      <td>33504580</td>\n",
       "      <td>33504580</td>\n",
       "      <td>Metabolic Codependencies in the Tumor Microenv...</td>\n",
       "      <td>Cancer discovery</td>\n",
       "      <td>Metabolic reprogramming enables cancer cell gr...</td>\n",
       "      <td>May-2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>27839715</td>\n",
       "      <td>27839715</td>\n",
       "      <td>27839715</td>\n",
       "      <td>27839715</td>\n",
       "      <td>[Lucy's cancer(s): A prehistorical origin?].</td>\n",
       "      <td>Gynecologie, obstetrique &amp; fertilite</td>\n",
       "      <td>The recent discovery of the earliest hominin c...</td>\n",
       "      <td>Dec-2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>PMID: 28232476</td>\n",
       "      <td>PMID:28232476</td>\n",
       "      <td>28232476</td>\n",
       "      <td>28232476</td>\n",
       "      <td>Potential Targets' Analysis Reveals Dual PI3K/...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td></td>\n",
       "      <td>Mar-2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>PMID: 27815673</td>\n",
       "      <td>PMID:27815673</td>\n",
       "      <td>27815673</td>\n",
       "      <td>27815673</td>\n",
       "      <td>Erratum to: Patient-Derived Mammosphere and Xe...</td>\n",
       "      <td>Journal of mammary gland biology and neoplasia</td>\n",
       "      <td></td>\n",
       "      <td>Dec-2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>PMID: 29079660</td>\n",
       "      <td>PMID:29079660</td>\n",
       "      <td>29079660</td>\n",
       "      <td>29079660</td>\n",
       "      <td>Dual mTOR Kinase Inhibitor MLN0128 Sensitizes HR</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td></td>\n",
       "      <td>Jan-2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>PMID:28473534</td>\n",
       "      <td>PMID:28473534</td>\n",
       "      <td>28473534</td>\n",
       "      <td>28473534</td>\n",
       "      <td>Elacestrant (RAD1901), a Selective Estrogen Re...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td></td>\n",
       "      <td>Aug-2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>PMID: 29093017</td>\n",
       "      <td>PMID:29093017</td>\n",
       "      <td>29093017</td>\n",
       "      <td>29093017</td>\n",
       "      <td>A Population of Heterogeneous Breast Cancer Pa...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td></td>\n",
       "      <td>Nov-2017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>131 rows × 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 27
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-28T09:01:32.889202Z",
     "start_time": "2024-06-28T09:01:05.169850Z"
    }
   },
   "cell_type": "code",
   "source": [
    "abstract_df = get_abstract_from_api(\"PMIDs_from_API.tsv\")\n",
    "abstract_df"
   ],
   "id": "ce4075d2ea062def",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           pubmed_ids type data_source  \\\n",
       "0      PMID: 19240712  PDX    Curie-BC   \n",
       "1      PMID: 21081655  PDX        PMLB   \n",
       "2      PMID: 21321221  PDX         LIH   \n",
       "3      PMID: 22247967  PDX    Curie-BC   \n",
       "4      PMID: 23460667  PDX         LIH   \n",
       "...               ...  ...         ...   \n",
       "3505    PMID:17606733  PDX    Curie-BC   \n",
       "3506    PMID:17606733  PDX    Curie-BC   \n",
       "3507    PMID:26440065  PDX        UMCG   \n",
       "3508    PMID:28522592  PDX    Curie-LC   \n",
       "3509    PMID:29458007  PDX        PMLB   \n",
       "\n",
       "                                      external_model_id            pid  \\\n",
       "0                                            ['HBCx-8']  PMID:19240712   \n",
       "1     ['PHLC412', 'PHLC344', 'PHLC134', 'PHLC432', '...  PMID:21081655   \n",
       "2                                                ['P3']  PMID:21321221   \n",
       "3                       ['HBCx-2', 'HBCx-31', 'HBCx-8']  PMID:22247967   \n",
       "4     ['T341', 'T331', 'T251', 'P3', 'P8', 'T16', 'T...  PMID:23460667   \n",
       "...                                                 ...            ...   \n",
       "3505  ['HBCx-5', 'HBCx-17', 'HBCx-3', 'HBCx-14', 'HB...  PMID:17606733   \n",
       "3506  ['HBCx-5', 'HBCx-17', 'HBCx-3', 'HBCx-14', 'HB...  PMID:17606733   \n",
       "3507  ['UMCGOVPDX68a', 'UMCGOVPDX70', 'UMCGOVPDX56',...  PMID:26440065   \n",
       "3508                          ['LCF9', 'LCF4', 'LCF15']  PMID:28522592   \n",
       "3509  ['PMLB3635', 'PMLB2345', 'PMLB2753', 'PMLB6130...  PMID:29458007   \n",
       "\n",
       "          pmid      PMID                                              Title  \\\n",
       "0     19240712  19240712  CD44 targeting reduces tumour growth and preve...   \n",
       "1     21081655  21081655  The ability to form primary tumor xenografts i...   \n",
       "2     21321221  21321221  Anti-VEGF treatment reduces blood supply and i...   \n",
       "3     22247967  22247967  Molecular profiling of patient-derived breast ...   \n",
       "4     23460667  23460667  Side population in human glioblastoma is non-t...   \n",
       "...        ...       ...                                                ...   \n",
       "3505  17606733  17606733  A new model of patient tumor-derived breast ca...   \n",
       "3506  17606733  17606733  A new model of patient tumor-derived breast ca...   \n",
       "3507  26440065  26440065  Biobanking of patient and patient-derived xeno...   \n",
       "3508  28522592  28522592  Sensitization of EGFR Wild-Type Non-Small Cell...   \n",
       "3509  29458007  29458007  A Genomically Characterized Collection of High...   \n",
       "\n",
       "                                                Journal  \\\n",
       "0                             British journal of cancer   \n",
       "1     Clinical cancer research : an official journal...   \n",
       "2     Proceedings of the National Academy of Science...   \n",
       "3                          Breast cancer research : BCR   \n",
       "4                        Brain : a journal of neurology   \n",
       "...                                                 ...   \n",
       "3505  Clinical cancer research : an official journal...   \n",
       "3506  Clinical cancer research : an official journal...   \n",
       "3507                                 Scientific reports   \n",
       "3508                      Molecular cancer therapeutics   \n",
       "3509                  The American journal of pathology   \n",
       "\n",
       "                                               Abstract      Date  \n",
       "0     CD44 is a marker of tumour-initiating cells an...  Mar-2009  \n",
       "1     Primary tumor xenografts (PTXG) established di...  Jan-2011  \n",
       "2     Bevacizumab, an antibody against vascular endo...  Mar-2011  \n",
       "3     Identification of new therapeutic agents for b...  Jan-2012  \n",
       "4     The identification and significance of cancer ...  May-2013  \n",
       "...                                                 ...       ...  \n",
       "3505  To establish a panel of human breast cancer (H...  Jul-2007  \n",
       "3506  To establish a panel of human breast cancer (H...  Jul-2007  \n",
       "3507  Using patient-derived xenografts (PDXs) for pr...  Oct-2015  \n",
       "3508  The benefit of EGFR-TKI in non-small cell lung...  Aug-2017  \n",
       "3509  High-grade serous ovarian cancer (HGSC) is the...  May-2018  \n",
       "\n",
       "[3510 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_ids</th>\n",
       "      <th>type</th>\n",
       "      <th>data_source</th>\n",
       "      <th>external_model_id</th>\n",
       "      <th>pid</th>\n",
       "      <th>pmid</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Title</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Abstract</th>\n",
       "      <th>Date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMID: 19240712</td>\n",
       "      <td>PDX</td>\n",
       "      <td>Curie-BC</td>\n",
       "      <td>['HBCx-8']</td>\n",
       "      <td>PMID:19240712</td>\n",
       "      <td>19240712</td>\n",
       "      <td>19240712</td>\n",
       "      <td>CD44 targeting reduces tumour growth and preve...</td>\n",
       "      <td>British journal of cancer</td>\n",
       "      <td>CD44 is a marker of tumour-initiating cells an...</td>\n",
       "      <td>Mar-2009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMID: 21081655</td>\n",
       "      <td>PDX</td>\n",
       "      <td>PMLB</td>\n",
       "      <td>['PHLC412', 'PHLC344', 'PHLC134', 'PHLC432', '...</td>\n",
       "      <td>PMID:21081655</td>\n",
       "      <td>21081655</td>\n",
       "      <td>21081655</td>\n",
       "      <td>The ability to form primary tumor xenografts i...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>Primary tumor xenografts (PTXG) established di...</td>\n",
       "      <td>Jan-2011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMID: 21321221</td>\n",
       "      <td>PDX</td>\n",
       "      <td>LIH</td>\n",
       "      <td>['P3']</td>\n",
       "      <td>PMID:21321221</td>\n",
       "      <td>21321221</td>\n",
       "      <td>21321221</td>\n",
       "      <td>Anti-VEGF treatment reduces blood supply and i...</td>\n",
       "      <td>Proceedings of the National Academy of Science...</td>\n",
       "      <td>Bevacizumab, an antibody against vascular endo...</td>\n",
       "      <td>Mar-2011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMID: 22247967</td>\n",
       "      <td>PDX</td>\n",
       "      <td>Curie-BC</td>\n",
       "      <td>['HBCx-2', 'HBCx-31', 'HBCx-8']</td>\n",
       "      <td>PMID:22247967</td>\n",
       "      <td>22247967</td>\n",
       "      <td>22247967</td>\n",
       "      <td>Molecular profiling of patient-derived breast ...</td>\n",
       "      <td>Breast cancer research : BCR</td>\n",
       "      <td>Identification of new therapeutic agents for b...</td>\n",
       "      <td>Jan-2012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMID: 23460667</td>\n",
       "      <td>PDX</td>\n",
       "      <td>LIH</td>\n",
       "      <td>['T341', 'T331', 'T251', 'P3', 'P8', 'T16', 'T...</td>\n",
       "      <td>PMID:23460667</td>\n",
       "      <td>23460667</td>\n",
       "      <td>23460667</td>\n",
       "      <td>Side population in human glioblastoma is non-t...</td>\n",
       "      <td>Brain : a journal of neurology</td>\n",
       "      <td>The identification and significance of cancer ...</td>\n",
       "      <td>May-2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3505</th>\n",
       "      <td>PMID:17606733</td>\n",
       "      <td>PDX</td>\n",
       "      <td>Curie-BC</td>\n",
       "      <td>['HBCx-5', 'HBCx-17', 'HBCx-3', 'HBCx-14', 'HB...</td>\n",
       "      <td>PMID:17606733</td>\n",
       "      <td>17606733</td>\n",
       "      <td>17606733</td>\n",
       "      <td>A new model of patient tumor-derived breast ca...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>To establish a panel of human breast cancer (H...</td>\n",
       "      <td>Jul-2007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3506</th>\n",
       "      <td>PMID:17606733</td>\n",
       "      <td>PDX</td>\n",
       "      <td>Curie-BC</td>\n",
       "      <td>['HBCx-5', 'HBCx-17', 'HBCx-3', 'HBCx-14', 'HB...</td>\n",
       "      <td>PMID:17606733</td>\n",
       "      <td>17606733</td>\n",
       "      <td>17606733</td>\n",
       "      <td>A new model of patient tumor-derived breast ca...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>To establish a panel of human breast cancer (H...</td>\n",
       "      <td>Jul-2007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3507</th>\n",
       "      <td>PMID:26440065</td>\n",
       "      <td>PDX</td>\n",
       "      <td>UMCG</td>\n",
       "      <td>['UMCGOVPDX68a', 'UMCGOVPDX70', 'UMCGOVPDX56',...</td>\n",
       "      <td>PMID:26440065</td>\n",
       "      <td>26440065</td>\n",
       "      <td>26440065</td>\n",
       "      <td>Biobanking of patient and patient-derived xeno...</td>\n",
       "      <td>Scientific reports</td>\n",
       "      <td>Using patient-derived xenografts (PDXs) for pr...</td>\n",
       "      <td>Oct-2015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3508</th>\n",
       "      <td>PMID:28522592</td>\n",
       "      <td>PDX</td>\n",
       "      <td>Curie-LC</td>\n",
       "      <td>['LCF9', 'LCF4', 'LCF15']</td>\n",
       "      <td>PMID:28522592</td>\n",
       "      <td>28522592</td>\n",
       "      <td>28522592</td>\n",
       "      <td>Sensitization of EGFR Wild-Type Non-Small Cell...</td>\n",
       "      <td>Molecular cancer therapeutics</td>\n",
       "      <td>The benefit of EGFR-TKI in non-small cell lung...</td>\n",
       "      <td>Aug-2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3509</th>\n",
       "      <td>PMID:29458007</td>\n",
       "      <td>PDX</td>\n",
       "      <td>PMLB</td>\n",
       "      <td>['PMLB3635', 'PMLB2345', 'PMLB2753', 'PMLB6130...</td>\n",
       "      <td>PMID:29458007</td>\n",
       "      <td>29458007</td>\n",
       "      <td>29458007</td>\n",
       "      <td>A Genomically Characterized Collection of High...</td>\n",
       "      <td>The American journal of pathology</td>\n",
       "      <td>High-grade serous ovarian cancer (HGSC) is the...</td>\n",
       "      <td>May-2018</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3510 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 18
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-17T10:32:50.270826Z",
     "start_time": "2024-06-17T10:26:43.991737Z"
    }
   },
   "cell_type": "code",
   "source": "abstract_df = read_pmid_get_abstract(\"pmid-cancer.tsv\")",
   "id": "e78518b1b31e0d74",
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mKeyboardInterrupt\u001B[0m                         Traceback (most recent call last)",
      "Cell \u001B[0;32mIn[8], line 1\u001B[0m\n\u001B[0;32m----> 1\u001B[0m abstract_df \u001B[38;5;241m=\u001B[39m \u001B[43mread_pmid_get_abstract\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mpmid-cancer.tsv\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m)\u001B[49m\n",
      "Cell \u001B[0;32mIn[7], line 8\u001B[0m, in \u001B[0;36mread_pmid_get_abstract\u001B[0;34m(path)\u001B[0m\n\u001B[1;32m      5\u001B[0m pmids \u001B[38;5;241m=\u001B[39m papers[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mpid\u001B[39m\u001B[38;5;124m'\u001B[39m]\u001B[38;5;241m.\u001B[39mstr\u001B[38;5;241m.\u001B[39mreplace(\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mPMID:\u001B[39m\u001B[38;5;124m'\u001B[39m, \u001B[38;5;124m'\u001B[39m\u001B[38;5;124m'\u001B[39m)\u001B[38;5;241m.\u001B[39mto_list()\n\u001B[1;32m      6\u001B[0m handle \u001B[38;5;241m=\u001B[39m Entrez\u001B[38;5;241m.\u001B[39mefetch(db\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mpubmed\u001B[39m\u001B[38;5;124m\"\u001B[39m, \u001B[38;5;28mid\u001B[39m\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124m,\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;241m.\u001B[39mjoin(\u001B[38;5;28mmap\u001B[39m(\u001B[38;5;28mstr\u001B[39m, pmids)),\n\u001B[1;32m      7\u001B[0m                        rettype\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mxml\u001B[39m\u001B[38;5;124m\"\u001B[39m, retmode\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mtext\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[0;32m----> 8\u001B[0m records \u001B[38;5;241m=\u001B[39m \u001B[43mEntrez\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[43mhandle\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m      9\u001B[0m abstracts \u001B[38;5;241m=\u001B[39m [(pubmed_article[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mMedlineCitation\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mPMID\u001B[39m\u001B[38;5;124m'\u001B[39m]\u001B[38;5;241m.\u001B[39mstrip(), pubmed_article[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mMedlineCitation\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mArticle\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mAbstract\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mAbstractText\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;241m0\u001B[39m]) \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mAbstract\u001B[39m\u001B[38;5;124m'\u001B[39m \u001B[38;5;129;01min\u001B[39;00m pubmed_article[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mMedlineCitation\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mArticle\u001B[39m\u001B[38;5;124m'\u001B[39m]\u001B[38;5;241m.\u001B[39mkeys() \u001B[38;5;28;01melse\u001B[39;00m (pubmed_article[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mMedlineCitation\u001B[39m\u001B[38;5;124m'\u001B[39m][\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mPMID\u001B[39m\u001B[38;5;124m'\u001B[39m]\u001B[38;5;241m.\u001B[39mstrip(), \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mNot provided\u001B[39m\u001B[38;5;124m'\u001B[39m)\n\u001B[1;32m     10\u001B[0m              \u001B[38;5;28;01mfor\u001B[39;00m pubmed_article \u001B[38;5;129;01min\u001B[39;00m records[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mPubmedArticle\u001B[39m\u001B[38;5;124m'\u001B[39m]]\n\u001B[1;32m     11\u001B[0m abstract_dict \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mdict\u001B[39m(abstracts)\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/Bio/Entrez/__init__.py:518\u001B[0m, in \u001B[0;36mread\u001B[0;34m(source, validate, escape, ignore_errors)\u001B[0m\n\u001B[1;32m    515\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mParser\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m DataHandler\n\u001B[1;32m    517\u001B[0m handler \u001B[38;5;241m=\u001B[39m DataHandler(validate, escape, ignore_errors)\n\u001B[0;32m--> 518\u001B[0m record \u001B[38;5;241m=\u001B[39m \u001B[43mhandler\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[43msource\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    519\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m record\n",
      "File \u001B[0;32m~/Library/Python/3.9/lib/python/site-packages/Bio/Entrez/Parser.py:403\u001B[0m, in \u001B[0;36mDataHandler.read\u001B[0;34m(self, source)\u001B[0m\n\u001B[1;32m    401\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mTypeError\u001B[39;00m(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mfile should be opened in binary mode\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[1;32m    402\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m--> 403\u001B[0m     \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mparser\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mParseFile\u001B[49m\u001B[43m(\u001B[49m\u001B[43mstream\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    404\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m expat\u001B[38;5;241m.\u001B[39mExpatError \u001B[38;5;28;01mas\u001B[39;00m e:\n\u001B[1;32m    405\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mparser\u001B[38;5;241m.\u001B[39mStartElementHandler:\n\u001B[1;32m    406\u001B[0m         \u001B[38;5;66;03m# We saw the initial <!xml declaration, so we can be sure that\u001B[39;00m\n\u001B[1;32m    407\u001B[0m         \u001B[38;5;66;03m# we are parsing XML data. Most likely, the XML file is\u001B[39;00m\n\u001B[1;32m    408\u001B[0m         \u001B[38;5;66;03m# corrupted.\u001B[39;00m\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:459\u001B[0m, in \u001B[0;36mHTTPResponse.read\u001B[0;34m(self, amt)\u001B[0m\n\u001B[1;32m    456\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m amt \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[1;32m    457\u001B[0m     \u001B[38;5;66;03m# Amount is given, implement using readinto\u001B[39;00m\n\u001B[1;32m    458\u001B[0m     b \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mbytearray\u001B[39m(amt)\n\u001B[0;32m--> 459\u001B[0m     n \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mreadinto\u001B[49m\u001B[43m(\u001B[49m\u001B[43mb\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    460\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28mmemoryview\u001B[39m(b)[:n]\u001B[38;5;241m.\u001B[39mtobytes()\n\u001B[1;32m    461\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m    462\u001B[0m     \u001B[38;5;66;03m# Amount is not given (unbounded read) so we must check self.length\u001B[39;00m\n\u001B[1;32m    463\u001B[0m     \u001B[38;5;66;03m# and self.chunked\u001B[39;00m\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:493\u001B[0m, in \u001B[0;36mHTTPResponse.readinto\u001B[0;34m(self, b)\u001B[0m\n\u001B[1;32m    490\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;241m0\u001B[39m\n\u001B[1;32m    492\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mchunked:\n\u001B[0;32m--> 493\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_readinto_chunked\u001B[49m\u001B[43m(\u001B[49m\u001B[43mb\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    495\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mlength \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[1;32m    496\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mlen\u001B[39m(b) \u001B[38;5;241m>\u001B[39m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mlength:\n\u001B[1;32m    497\u001B[0m         \u001B[38;5;66;03m# clip the read to the \"end of response\"\u001B[39;00m\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:588\u001B[0m, in \u001B[0;36mHTTPResponse._readinto_chunked\u001B[0;34m(self, b)\u001B[0m\n\u001B[1;32m    586\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m    587\u001B[0m     \u001B[38;5;28;01mwhile\u001B[39;00m \u001B[38;5;28;01mTrue\u001B[39;00m:\n\u001B[0;32m--> 588\u001B[0m         chunk_left \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_get_chunk_left\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    589\u001B[0m         \u001B[38;5;28;01mif\u001B[39;00m chunk_left \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[1;32m    590\u001B[0m             \u001B[38;5;28;01mreturn\u001B[39;00m total_bytes\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:556\u001B[0m, in \u001B[0;36mHTTPResponse._get_chunk_left\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m    554\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_safe_read(\u001B[38;5;241m2\u001B[39m)  \u001B[38;5;66;03m# toss the CRLF at the end of the chunk\u001B[39;00m\n\u001B[1;32m    555\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m--> 556\u001B[0m     chunk_left \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_read_next_chunk_size\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    557\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m:\n\u001B[1;32m    558\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m IncompleteRead(\u001B[38;5;124mb\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124m'\u001B[39m)\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:516\u001B[0m, in \u001B[0;36mHTTPResponse._read_next_chunk_size\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m    514\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21m_read_next_chunk_size\u001B[39m(\u001B[38;5;28mself\u001B[39m):\n\u001B[1;32m    515\u001B[0m     \u001B[38;5;66;03m# Read the next chunk size from the file\u001B[39;00m\n\u001B[0;32m--> 516\u001B[0m     line \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mfp\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mreadline\u001B[49m\u001B[43m(\u001B[49m\u001B[43m_MAXLINE\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m+\u001B[39;49m\u001B[43m \u001B[49m\u001B[38;5;241;43m1\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[1;32m    517\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mlen\u001B[39m(line) \u001B[38;5;241m>\u001B[39m _MAXLINE:\n\u001B[1;32m    518\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m LineTooLong(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mchunk size\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/socket.py:704\u001B[0m, in \u001B[0;36mSocketIO.readinto\u001B[0;34m(self, b)\u001B[0m\n\u001B[1;32m    702\u001B[0m \u001B[38;5;28;01mwhile\u001B[39;00m \u001B[38;5;28;01mTrue\u001B[39;00m:\n\u001B[1;32m    703\u001B[0m     \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m--> 704\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_sock\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mrecv_into\u001B[49m\u001B[43m(\u001B[49m\u001B[43mb\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    705\u001B[0m     \u001B[38;5;28;01mexcept\u001B[39;00m timeout:\n\u001B[1;32m    706\u001B[0m         \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_timeout_occurred \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;01mTrue\u001B[39;00m\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1241\u001B[0m, in \u001B[0;36mSSLSocket.recv_into\u001B[0;34m(self, buffer, nbytes, flags)\u001B[0m\n\u001B[1;32m   1237\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m flags \u001B[38;5;241m!=\u001B[39m \u001B[38;5;241m0\u001B[39m:\n\u001B[1;32m   1238\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m(\n\u001B[1;32m   1239\u001B[0m           \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mnon-zero flags not allowed in calls to recv_into() on \u001B[39m\u001B[38;5;132;01m%s\u001B[39;00m\u001B[38;5;124m\"\u001B[39m \u001B[38;5;241m%\u001B[39m\n\u001B[1;32m   1240\u001B[0m           \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m\u001B[38;5;18m__class__\u001B[39m)\n\u001B[0;32m-> 1241\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[43mnbytes\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mbuffer\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m   1242\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m   1243\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28msuper\u001B[39m()\u001B[38;5;241m.\u001B[39mrecv_into(buffer, nbytes, flags)\n",
      "File \u001B[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1099\u001B[0m, in \u001B[0;36mSSLSocket.read\u001B[0;34m(self, len, buffer)\u001B[0m\n\u001B[1;32m   1097\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m   1098\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m buffer \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[0;32m-> 1099\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_sslobj\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;28;43mlen\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mbuffer\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m   1100\u001B[0m     \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m   1101\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_sslobj\u001B[38;5;241m.\u001B[39mread(\u001B[38;5;28mlen\u001B[39m)\n",
      "\u001B[0;31mKeyboardInterrupt\u001B[0m: "
     ]
    }
   ],
   "execution_count": 8
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-21T11:56:52.221122Z",
     "start_time": "2024-06-21T11:56:50.434633Z"
    }
   },
   "cell_type": "code",
   "source": [
    "publications = pd.read_json(\"https://www.cancermodels.org/api/publication_group\")\n",
    "model_information = pd.read_json(\"https://www.cancermodels.org/api/model_information?select=external_model_id,type,data_source,publication_group_id\")"
   ],
   "id": "5d1b9922e8a1b9ff",
   "outputs": [],
   "execution_count": 9
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-21T12:05:37.507950Z",
     "start_time": "2024-06-21T12:05:37.250167Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df = model_information.merge(publications, left_on='publication_group_id', right_on='id', how='left')[['external_model_id','type','data_source','pubmed_ids']]\n",
    "df = df[~df['pubmed_ids'].isna()].reset_index(drop=True)\n",
    "df['pubmed_ids'] = df['pubmed_ids'].str.split(',')\n",
    "df = df.explode('pubmed_ids')\n",
    "df = df[df['pubmed_ids'] != \"\"].reset_index(drop=True)\n",
    "df = df.groupby(['pubmed_ids', 'type', 'data_source'])['external_model_id'].apply(list).reset_index()"
   ],
   "id": "d55878ef2863bbff",
   "outputs": [],
   "execution_count": 28
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-21T12:06:05.959491Z",
     "start_time": "2024-06-21T12:06:05.707950Z"
    }
   },
   "cell_type": "code",
   "source": "df.to_csv('PMIDs_from_API.tsv', sep='\\t', index=False)",
   "id": "55075a121992fb51",
   "outputs": [],
   "execution_count": 29
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-28T09:00:47.047418Z",
     "start_time": "2024-06-28T09:00:46.316087Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from requests import get\n",
    "import xml.etree.ElementTree as ET\n",
    "\n",
    "response = get(\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=36264037,4357757,4353636\")\n",
    "# Parse the XML content\n",
    "root = ET.fromstring(response.text)\n",
    "\n",
    "# Extract information\n",
    "articles = []\n",
    "for article in root.findall('PubmedArticle'):\n",
    "    pmid = article.find('.//PMID').text\n",
    "    title = article.find('.//ArticleTitle').text\n",
    "    journal_title = article.find('.//Journal//Title').text\n",
    "    abstract = article.findall('.//AbstractText')\n",
    "    abstract_text = \"\"\n",
    "    for abs in abstract:\n",
    "        text = ET.tostring(abs, encoding='unicode', method='text')\n",
    "        abstract_text += replace_html_tags(text)\n",
    "    pub_date_element = article.find('.//DateCompleted')\n",
    "    year = pub_date_element.find('Year').text if pub_date_element.find('Year') is not None else ''\n",
    "    month = pub_date_element.find('Month').text if pub_date_element.find('Month') is not None else ''\n",
    "    publication_date = f\"{month}-{year}\"\n",
    "    articles.append({\n",
    "        'PMID': pmid,\n",
    "        'Title': title,\n",
    "        'Journal': journal_title,\n",
    "        'Abstract': abstract_text,\n",
    "        'Date': publication_date\n",
    "        \n",
    "    })\n",
    "articles"
   ],
   "id": "7bf989c9a0a122d",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'PMID': '36264037',\n",
       "  'Title': 'Chemical proteomics reveals interactors of the alarmone diadenosine triphosphate in the cancer cell line H1299.',\n",
       "  'Journal': 'Journal of peptide science : an official publication of the European Peptide Society',\n",
       "  'Abstract': 'Intracellular dinucleoside polyphosphates (Npn Ns) have been known for decades but the functional role remains enigmatic. Diadenosine triphosphate (Ap3 A) is one of the most prominent examples, and its intercellular concentration was shown to increase upon cellular stress. By employment of previously reported Ap3 A-based photoaffinity-labeling probes (PALPs) in chemical proteomics, we investigated the Ap3 A interactome in the human lung carcinoma cell line H1299. The cell line is deficient of the fragile histidine triade (Fhit) protein, a hydrolase of Ap3 A and tumor suppressor. Overall, the number of identified potential interaction partners was significantly lower than in the previously investigated HEK293T cell line. Gene ontology term analysis revealed that the identified proteins participate in similar pathways as for HEK293T, but the percentage of proteins involved in RNA-related processes is higher for H1299. The obtained results highlight similarities and differences of the Ap3 A interaction network in different cell lines and give further indications regarding the importance of the presence of Fhit.',\n",
       "  'Date': '02-2023'},\n",
       " {'PMID': '4357757',\n",
       "  'Title': 'A human cell line from a pleural effusion derived from a breast carcinoma.',\n",
       "  'Journal': 'Journal of the National Cancer Institute',\n",
       "  'Abstract': '',\n",
       "  'Date': '02-1974'},\n",
       " {'PMID': '4353636',\n",
       "  'Title': 'Estrogen receptor in a human cell line (MCF-7) from breast carcinoma.',\n",
       "  'Journal': 'The Journal of biological chemistry',\n",
       "  'Abstract': '',\n",
       "  'Date': '10-1973'}]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-06-24T10:12:08.646479Z",
     "start_time": "2024-06-24T10:12:08.641841Z"
    }
   },
   "cell_type": "code",
   "source": "article.findall('.//AbstractText')",
   "id": "832459bf6093b830",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<Element 'AbstractText' at 0x13e347680>]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 113
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-19T13:17:54.696464Z",
     "start_time": "2024-07-19T13:17:51.198482Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from requests import get\n",
    "import xml.etree.ElementTree as ET\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "def check_keywords(text, keywords):\n",
    "    return any(keyword.lower() in text.lower() for keyword in keywords)\n",
    "\n",
    "def get_article_details(papers):\n",
    "    ids = ','.join(papers['pmid'].to_list())\n",
    "    payload = {'db':'pubmed', 'id':ids}\n",
    "    response = post(f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi\", data=payload)\n",
    "    root = ET.fromstring(response.text)\n",
    "    articles = []\n",
    "    for article in root.findall('PubmedArticle'):\n",
    "        pmid = article.find('.//PMID').text\n",
    "        title = article.find('.//ArticleTitle').text\n",
    "        journal_title = article.find('.//Journal//Title').text\n",
    "        abstract = article.findall('.//AbstractText')\n",
    "        abstract_text = \"\"\n",
    "        for abs in abstract:\n",
    "            if abs.text is not None:\n",
    "                text = ET.tostring(abs, encoding='unicode', method='text')\n",
    "                abstract_text += replace_html_tags(text)\n",
    "        pub_date_element = article.find('.//JournalIssue/PubDate')\n",
    "        year = pub_date_element.find('Year').text if pub_date_element.find('Year') is not None else ''\n",
    "        month = pub_date_element.find('Month').text if pub_date_element.find('Month') is not None else ''\n",
    "        publication_date = f\"{month}-{year}\"\n",
    "        articles.append({\n",
    "            'PMID': pmid,\n",
    "            'Title': title,\n",
    "            'Journal': journal_title,\n",
    "            'Abstract': abstract_text,\n",
    "            'Date': publication_date\n",
    "            \n",
    "        })\n",
    "    abstract_df = pd.DataFrame(articles)\n",
    "    abstract_df = papers.merge(abstract_df, left_on='pmid', right_on='PMID', how='left')[['filename','PubMedID', 'Title', 'Journal', 'Date', 'abstract', 'pmid']]\n",
    "    xenograft_keywords = ['xenograft', 'PDX']\n",
    "    organoid_keywords = ['organoid', 'PDO']\n",
    "    cell_keywords = ['cell line']\n",
    "    abstract_df['xenograft_pdx'] = abstract_df['abstract'].apply(lambda x: 'PDX' if check_keywords(x, xenograft_keywords) else '')\n",
    "    abstract_df['organoid_pdo'] = abstract_df['abstract'].apply(lambda x: 'Organoid' if check_keywords(x, organoid_keywords) else '')\n",
    "    abstract_df['cell_line_pdo'] = abstract_df['abstract'].apply(lambda x: 'Cell line' if check_keywords(x, cell_keywords) else '')\n",
    "    \n",
    "    abstract_df['stratify'] = abstract_df['xenograft_pdx'] + '_' + abstract_df['organoid_pdo']\n",
    "    train, temp = train_test_split(abstract_df, test_size=0.4, stratify=abstract_df['stratify'], random_state=42)\n",
    "    dev, test = train_test_split(temp, test_size=0.5, stratify=temp['stratify'], random_state=42)\n",
    "    train['split'] = 'train'\n",
    "    dev['split'] = 'dev'\n",
    "    test['split'] = 'test'\n",
    "    df_split = pd.concat([train, dev, test]).reset_index(drop=True)\n",
    "    df_split = df_split.drop(columns=['stratify'])\n",
    "    return df_split\n",
    "\n",
    "corpus_path = \"corpus-txt\"\n",
    "corpus_files = [join(corpus_path, f) for f in get_files(corpus_path) if f.endswith('.txt')]\n",
    "corpus_df = pd.DataFrame()\n",
    "temp = pd.DataFrame()\n",
    "for f in corpus_files:\n",
    "    d = open(f, \"r\")\n",
    "    txt = d.read()\n",
    "    temp['PubMedID'] = [f.replace(corpus_path, '').replace('/', '').replace('.txt', '').replace('_', ':')]\n",
    "    temp['filename'] = f.replace(corpus_path, '').replace('/', '')\n",
    "    temp['pmid'] = f.replace(corpus_path, '').replace('/', '').replace('.txt', '').replace('_', ':').replace('PMID:', '')\n",
    "    temp['abstract'] = txt\n",
    "    corpus_df = pd.concat([corpus_df, temp])\n",
    "\n",
    "detailed_df = get_article_details(corpus_df)\n",
    "detailed_df.to_csv('corpus_abtract_detailed.tsv', index=False, sep='\\t')\n",
    "detailed_df\n",
    "#abstract_df.to_csv('corpus_abtract_detailed.tsv', index=False, sep='\\t')"
   ],
   "id": "fc73917f2110d66f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             filename       PubMedID  \\\n",
       "0   PMID_30146332.txt  PMID:30146332   \n",
       "1   PMID_32553164.txt  PMID:32553164   \n",
       "2   PMID_34020697.txt  PMID:34020697   \n",
       "3   PMID_29242316.txt  PMID:29242316   \n",
       "4   PMID_25863122.txt  PMID:25863122   \n",
       "..                ...            ...   \n",
       "95  PMID_37452026.txt  PMID:37452026   \n",
       "96  PMID_36657446.txt  PMID:36657446   \n",
       "97  PMID_32183023.txt  PMID:32183023   \n",
       "98  PMID_28232476.txt  PMID:28232476   \n",
       "99  PMID_36223547.txt  PMID:36223547   \n",
       "\n",
       "                                                Title  \\\n",
       "0   Identification of Therapeutic Targets in Rhabd...   \n",
       "1   Patient-Derived Ovarian Cancer Organoids Mimic...   \n",
       "2   Analysis of genomic and non-genomic signaling ...   \n",
       "3   Colorectal Cancer Consensus Molecular Subtypes...   \n",
       "4   Development and characterization of a human or...   \n",
       "..                                                ...   \n",
       "95  Oxidative phosphorylation is a metabolic vulne...   \n",
       "96  Using patient-derived organoids to predict loc...   \n",
       "97  The Second Generation Antibody-Drug Conjugate ...   \n",
       "98  Potential Targets' Analysis Reveals Dual PI3K/...   \n",
       "99  Tinengotinib (TT-00420), a Novel Spectrum-Sele...   \n",
       "\n",
       "                                              Journal      Date  \\\n",
       "0                                         Cancer cell  Sep-2018   \n",
       "1                                        Cell reports  Jun-2020   \n",
       "2                        Breast cancer research : BCR  May-2021   \n",
       "3   Clinical cancer research : an official journal...  Feb-2018   \n",
       "4                               Developmental biology  Nov-2015   \n",
       "..                                                ...       ...   \n",
       "95                              Nature communications  Jul-2023   \n",
       "96                             Cell reports. Medicine  Feb-2023   \n",
       "97                                            Cancers  Mar-2020   \n",
       "98  Clinical cancer research : an official journal...  Mar-2017   \n",
       "99                      Molecular cancer therapeutics  Feb-2023   \n",
       "\n",
       "                                             abstract      pmid xenograft_pdx  \\\n",
       "0   Personalized cancer therapy targeting somatic ...  30146332                 \n",
       "1   There remains an unmet need for preclinical mo...  32553164                 \n",
       "2   Endocrine therapies targeting estrogen signali...  34020697           PDX   \n",
       "3   Purpose: Response to standard oncologic treatm...  29242316           PDX   \n",
       "4   Neuroblastoma is a pediatric cancer of the dev...  25863122           PDX   \n",
       "..                                                ...       ...           ...   \n",
       "95  Resistance to endocrine treatments and CDK4/6 ...  37452026           PDX   \n",
       "96  Predicting the clinical response to chemothera...  36657446                 \n",
       "97  Trastuzumab-emtansine (T-DM1) is an antibody-d...  32183023                 \n",
       "98  Purpose: Uterine sarcomas are rare and heterog...  28232476           PDX   \n",
       "99  Triple-negative breast cancer (TNBC) is a high...  36223547                 \n",
       "\n",
       "   organoid_pdo cell_line_pdo  split  \n",
       "0                              train  \n",
       "1      Organoid                train  \n",
       "2                              train  \n",
       "3                   Cell line  train  \n",
       "4                              train  \n",
       "..          ...           ...    ...  \n",
       "95                  Cell line   test  \n",
       "96     Organoid                 test  \n",
       "97                              test  \n",
       "98                              test  \n",
       "99                  Cell line   test  \n",
       "\n",
       "[100 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>PubMedID</th>\n",
       "      <th>Title</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Date</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pmid</th>\n",
       "      <th>xenograft_pdx</th>\n",
       "      <th>organoid_pdo</th>\n",
       "      <th>cell_line_pdo</th>\n",
       "      <th>split</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMID_30146332.txt</td>\n",
       "      <td>PMID:30146332</td>\n",
       "      <td>Identification of Therapeutic Targets in Rhabd...</td>\n",
       "      <td>Cancer cell</td>\n",
       "      <td>Sep-2018</td>\n",
       "      <td>Personalized cancer therapy targeting somatic ...</td>\n",
       "      <td>30146332</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMID_32553164.txt</td>\n",
       "      <td>PMID:32553164</td>\n",
       "      <td>Patient-Derived Ovarian Cancer Organoids Mimic...</td>\n",
       "      <td>Cell reports</td>\n",
       "      <td>Jun-2020</td>\n",
       "      <td>There remains an unmet need for preclinical mo...</td>\n",
       "      <td>32553164</td>\n",
       "      <td></td>\n",
       "      <td>Organoid</td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMID_34020697.txt</td>\n",
       "      <td>PMID:34020697</td>\n",
       "      <td>Analysis of genomic and non-genomic signaling ...</td>\n",
       "      <td>Breast cancer research : BCR</td>\n",
       "      <td>May-2021</td>\n",
       "      <td>Endocrine therapies targeting estrogen signali...</td>\n",
       "      <td>34020697</td>\n",
       "      <td>PDX</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMID_29242316.txt</td>\n",
       "      <td>PMID:29242316</td>\n",
       "      <td>Colorectal Cancer Consensus Molecular Subtypes...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>Feb-2018</td>\n",
       "      <td>Purpose: Response to standard oncologic treatm...</td>\n",
       "      <td>29242316</td>\n",
       "      <td>PDX</td>\n",
       "      <td></td>\n",
       "      <td>Cell line</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMID_25863122.txt</td>\n",
       "      <td>PMID:25863122</td>\n",
       "      <td>Development and characterization of a human or...</td>\n",
       "      <td>Developmental biology</td>\n",
       "      <td>Nov-2015</td>\n",
       "      <td>Neuroblastoma is a pediatric cancer of the dev...</td>\n",
       "      <td>25863122</td>\n",
       "      <td>PDX</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>PMID_37452026.txt</td>\n",
       "      <td>PMID:37452026</td>\n",
       "      <td>Oxidative phosphorylation is a metabolic vulne...</td>\n",
       "      <td>Nature communications</td>\n",
       "      <td>Jul-2023</td>\n",
       "      <td>Resistance to endocrine treatments and CDK4/6 ...</td>\n",
       "      <td>37452026</td>\n",
       "      <td>PDX</td>\n",
       "      <td></td>\n",
       "      <td>Cell line</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>PMID_36657446.txt</td>\n",
       "      <td>PMID:36657446</td>\n",
       "      <td>Using patient-derived organoids to predict loc...</td>\n",
       "      <td>Cell reports. Medicine</td>\n",
       "      <td>Feb-2023</td>\n",
       "      <td>Predicting the clinical response to chemothera...</td>\n",
       "      <td>36657446</td>\n",
       "      <td></td>\n",
       "      <td>Organoid</td>\n",
       "      <td></td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>PMID_32183023.txt</td>\n",
       "      <td>PMID:32183023</td>\n",
       "      <td>The Second Generation Antibody-Drug Conjugate ...</td>\n",
       "      <td>Cancers</td>\n",
       "      <td>Mar-2020</td>\n",
       "      <td>Trastuzumab-emtansine (T-DM1) is an antibody-d...</td>\n",
       "      <td>32183023</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>PMID_28232476.txt</td>\n",
       "      <td>PMID:28232476</td>\n",
       "      <td>Potential Targets' Analysis Reveals Dual PI3K/...</td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>Mar-2017</td>\n",
       "      <td>Purpose: Uterine sarcomas are rare and heterog...</td>\n",
       "      <td>28232476</td>\n",
       "      <td>PDX</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>PMID_36223547.txt</td>\n",
       "      <td>PMID:36223547</td>\n",
       "      <td>Tinengotinib (TT-00420), a Novel Spectrum-Sele...</td>\n",
       "      <td>Molecular cancer therapeutics</td>\n",
       "      <td>Feb-2023</td>\n",
       "      <td>Triple-negative breast cancer (TNBC) is a high...</td>\n",
       "      <td>36223547</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Cell line</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-19T13:01:52.708179Z",
     "start_time": "2024-07-19T13:01:52.702885Z"
    }
   },
   "cell_type": "code",
   "source": "detailed_df.abstract.str.contains('PDX')",
   "id": "7c7c82a2c5a6d34e",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['PubMedID', 'pmid', 'abstract', 'PMID', 'Title', 'Journal', 'Abstract',\n",
       "       'Date'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 27
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T13:36:45.750048Z",
     "start_time": "2024-10-30T13:36:39.591158Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import os\n",
    "import json\n",
    "from requests import post, exceptions, get\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "\n",
    "def pmid_to_pmcid(pmids):\n",
    "    pmc_ids = {}\n",
    "    \n",
    "    for pmid in pmids:\n",
    "        try:\n",
    "            # Fetch metadata for the given PMID\n",
    "            response = get(f\"https://www.ebi.ac.uk/europepmc/webservices/rest/search?query=EXT_ID:{pmid}&format=json\")\n",
    "            response.raise_for_status()\n",
    "            data = response.json()\n",
    "            \n",
    "            # Check if the data contains results and a PMCID\n",
    "            if data.get(\"resultList\", {}).get(\"result\"):\n",
    "                result = data[\"resultList\"][\"result\"][0]\n",
    "                pmcid = result.get(\"pmcid\")\n",
    "                \n",
    "                # Store in dictionary\n",
    "                pmc_ids[pmid] = str(pmcid) if pmcid else \"\"\n",
    "            else:\n",
    "                pmc_ids[pmid] = ''\n",
    "        \n",
    "        except exceptions.RequestException as e:\n",
    "            print(f\"Error retrieving PMCID for PMID {pmid}: {e}\")\n",
    "            pmc_ids[pmid] = \"Error occurred\"\n",
    "    \n",
    "    return pmc_ids\n",
    "\n",
    "def fetch_full_text(pmids, output_folder=\"articles\"):\n",
    "    # Ensure output folder exists\n",
    "    os.makedirs(output_folder, exist_ok=True)\n",
    "    \n",
    "    for pmid in pmids:\n",
    "        if pmid == '':\n",
    "            continue\n",
    "        try:\n",
    "            # Fetch article details\n",
    "            response = get(f\"https://www.ebi.ac.uk/europepmc/webservices/rest/{pmid}/fullTextXML\")\n",
    "            response.raise_for_status()\n",
    "            \n",
    "            if response.text:\n",
    "                xml_content = response.text\n",
    "                \n",
    "                # Parse XML content with BeautifulSoup\n",
    "                soup = BeautifulSoup(xml_content, \"xml\")\n",
    "                \n",
    "                # Extract and concatenate all text elements\n",
    "                article_text = \" \".join([p.get_text() for p in soup.find('body').find_all(\"p\")])\n",
    "                \n",
    "                # Save parsed text to file\n",
    "                file_path = os.path.join(output_folder, f\"{pmid}.txt\")\n",
    "                with open(file_path, \"w\", encoding=\"utf-8\") as file:\n",
    "                    file.write(article_text)\n",
    "                \n",
    "                \n",
    "                print(f\"Full text for PMID {pmid} saved successfully.\")\n",
    "            else:\n",
    "                print(f\"No full text available for PMID {pmid}.\")\n",
    "        \n",
    "        except exceptions.RequestException as e:\n",
    "            print(f\"Error retrieving PMID {pmid}: {e}\")\n",
    "    return xml_content\n",
    "\n",
    "def read_in_full_text(input_dir):\n",
    "    full_id2text = {}\n",
    "    for fl in os.listdir(input_dir):\n",
    "        abstract_id = re.search(r\"PMC\\d+\", fl, flags=0).group()\n",
    "        with open(os.path.join(input_dir, fl), \"r\") as fr:\n",
    "            text = fr.read()\n",
    "        full_id2text[abstract_id] = text\n",
    "    return full_id2text\n",
    "\n",
    "\n",
    "def tokenize_full_text(fulltext_dir):\n",
    "    full_texts = read_in_full_text(fulltext_dir)\n",
    "    tokenized_full_texts = {}\n",
    "    for id, text in full_texts.items():\n",
    "        tokenized = text.split('. ')\n",
    "        tokenized_full_text = get_sentence_character_positions(tokenized)\n",
    "        tokenized_full_texts[id] = tokenized_full_text\n",
    "    return tokenized_full_texts\n",
    "    \n",
    "        \n",
    "    \n",
    "def get_sentence_character_positions(sentences):\n",
    "    result = []\n",
    "    char_start = 0  # Initialize character start position\n",
    "\n",
    "    for idx, sentence in enumerate(sentences):\n",
    "        char_end = char_start + len(sentence)  # Calculate character end position\n",
    "        # Create the sentence dictionary\n",
    "        sentence_info = {\n",
    "            \"sent_idx\": idx,  # Indexing starts from 1\n",
    "            \"char_start\": char_start,\n",
    "            \"char_end\": char_end,\n",
    "            \"sent_text\": sentence\n",
    "        }\n",
    "        result.append(sentence_info)  # Add to result list\n",
    "        char_start = char_end + 1  # Update character start for next sentence\n",
    "\n",
    "    return result\n",
    "\n",
    "def save_sentences_to_json(sentences_with_positions, output_file):\n",
    "    with open(output_file, 'w') as json_file:\n",
    "        json.dump(sentences_with_positions, json_file, indent=4)\n",
    "\n",
    "# Example list of PMIDs\n",
    "pmid_list = [pmid.replace('.txt', '').replace('PMID_', '') for pmid in os.listdir('/Users/tushar/CancerModels/set-aside/sa-y4/prompt_pdcr_ebi/abstract_texts_updated_split/dev') if pmid.__contains__('PMID')]\n",
    "pmcid_list = pmid_to_pmcid(pmid_list)\n",
    "xml = fetch_full_text(pmcid_list.values())\n",
    "tokenized = tokenize_full_text('articles/')\n",
    "save_sentences_to_json(tokenized, 'fulltext_tokenized_dev.json')"
   ],
   "id": "db862f405ead80f0",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Full text for PMID PMC4815803 saved successfully.\n",
      "Full text for PMID PMC9631073 saved successfully.\n",
      "Error retrieving PMID PMC7556703: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC7556703/fullTextXML\n",
      "Error retrieving PMID PMC4301580: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC4301580/fullTextXML\n",
      "Full text for PMID PMC8132120 saved successfully.\n",
      "Error retrieving PMID PMC8662740: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC8662740/fullTextXML\n",
      "Full text for PMID PMC5499209 saved successfully.\n",
      "Error retrieving PMID PMC6342199: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC6342199/fullTextXML\n",
      "Full text for PMID PMC7471705 saved successfully.\n",
      "Full text for PMID PMC7612638 saved successfully.\n",
      "Error retrieving PMID PMC7350550: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC7350550/fullTextXML\n",
      "Error retrieving PMID PMC6158019: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC6158019/fullTextXML\n",
      "Full text for PMID PMC8134568 saved successfully.\n",
      "Full text for PMID PMC9270001 saved successfully.\n",
      "Error retrieving PMID PMC9177814: 404 Client Error: Not Found for url: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC9177814/fullTextXML\n"
     ]
    }
   ],
   "execution_count": 59
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T13:36:20.111507Z",
     "start_time": "2024-10-30T13:36:20.102698Z"
    }
   },
   "cell_type": "code",
   "source": "read_in_full_text('articles/').keys()",
   "id": "c243c2f85db87748",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['PMC9631073', 'PMC7612638', 'PMC4815803', 'PMC5499209', 'PMC8132120', 'PMC8134568', 'PMC7471705', 'PMC9270001'])"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 58
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T13:12:44.779018Z",
     "start_time": "2024-10-30T13:12:44.771227Z"
    }
   },
   "cell_type": "code",
   "source": "re.search(r\"PMC\\d+\", fl, flags=0)",
   "id": "9d7b3ea79cf443bf",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<re.Match object; span=(0, 10), match='PMC5499209'>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 40
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T15:19:45.361614Z",
     "start_time": "2024-10-30T15:19:45.350880Z"
    }
   },
   "cell_type": "code",
   "source": "pmcid_list",
   "id": "520cc47bb8ac1600",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'26695443': 'PMC4815803',\n",
       " '36058001': 'PMC9631073',\n",
       " '33421710': '',\n",
       " '31761724': '',\n",
       " '31883794': 'PMC7556703',\n",
       " '25444907': 'PMC4301580',\n",
       " '34027491': 'PMC8132120',\n",
       " '32075943': 'PMC8662740',\n",
       " '28561063': 'PMC5499209',\n",
       " '30395907': 'PMC6342199',\n",
       " '32884042': 'PMC7471705',\n",
       " '35410383': 'PMC7612638',\n",
       " '32294323': 'PMC7350550',\n",
       " '30146332': 'PMC6158019',\n",
       " '34011980': 'PMC8134568',\n",
       " '29458007': '',\n",
       " '35802820': 'PMC9270001',\n",
       " '30380421': '',\n",
       " '35508177': 'PMC9177814',\n",
       " '30017245': ''}"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 64
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "6b353a7ceba77065"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T15:09:39.413617Z",
     "start_time": "2024-10-30T15:09:39.392243Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "\n",
    "# Load JSON data\n",
    "with open('/Users/tushar/CancerModels/set-aside/sa-y4/prompt_pdcr_ebi/gpt-4o_dev_output_parsed.json', 'r') as file:\n",
    "    data = json.load(file)\n",
    "\n",
    "# Prepare a dictionary to hold lists of tags for each PMID\n",
    "df_dict = defaultdict(lambda: defaultdict(list))\n",
    "\n",
    "# Populate dictionary by iterating over each PMID\n",
    "for pmid, annotations in data.items():\n",
    "    for annotation in annotations:\n",
    "        tag = annotation['tags'][0]  # Take the first tag from the list\n",
    "        text_provided = annotation['textProvided']\n",
    "        df_dict[pmid][tag].append(text_provided)  # Append text under the tag column for each PMID\n",
    "\n",
    "# Convert to DataFrame\n",
    "df = pd.DataFrame({pmid: {k: v for k, v in tags.items()} for pmid, tags in df_dict.items()}).T\n",
    "\n",
    "# Optional: Join lists into comma-separated strings\n",
    "#df = df.applymap(lambda x: ', '.join(x) if isinstance(x, list) else x)\n",
    "\n",
    "df.to_csv('parsed_llm_output_gpt4o_combined.tsv', sep='\\t')\n"
   ],
   "id": "3e72fb0c9b440d11",
   "outputs": [],
   "execution_count": 63
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "fa65f5e64b96812b"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T11:35:53.452700Z",
     "start_time": "2024-10-30T11:35:53.359289Z"
    }
   },
   "cell_type": "code",
   "source": "soup = BeautifulSoup(xml, \"xml\")",
   "id": "13292b46d9111a7c",
   "outputs": [],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T11:37:19.525416Z",
     "start_time": "2024-10-30T11:37:19.520293Z"
    }
   },
   "cell_type": "code",
   "source": "body = soup.find('body')",
   "id": "a8aa380ca2e54fad",
   "outputs": [],
   "execution_count": 20
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T13:03:55.465690Z",
     "start_time": "2024-10-30T13:03:55.459583Z"
    }
   },
   "cell_type": "code",
   "source": [
    "for token in body.find_all('p')[0].get_text().split('. '):\n",
    "    print(len(token))"
   ],
   "id": "3690abda951649c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "182\n",
      "233\n",
      "265\n"
     ]
    }
   ],
   "execution_count": 31
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T13:06:48.222744Z",
     "start_time": "2024-10-30T13:06:48.212198Z"
    }
   },
   "cell_type": "code",
   "source": "body.find_all('p')[0].get_text().split('. ')",
   "id": "c87e768b7adfce82",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A number of classification systems based on gene expression have been proposed that stratify colorectal cancer (CRC) in subgroups with distinct molecular and clinical features1234567',\n",
       " 'Comparative analyses in different data sets have revealed substantial classification coherence across the various signatures, particularly in the case of a ‘Stem/Serrated/Mesenchymal’ (SSM) subtype endowed with negative prognosis8910',\n",
       " 'These classification efforts have been recently consolidated by a multi-institutional initiative that comprehensively cross compared the different subtype assignments on a common set of samples, leading to the definition of the consensus molecular subtypes11 (CMS).']"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 32
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-12-11T11:16:07.336359Z",
     "start_time": "2024-12-11T11:16:06.452700Z"
    }
   },
   "cell_type": "code",
   "source": [
    "pmids_positive = ['32553164', '34020697', '29242316', '25863122', '28854174', '30146162', '37610680', '29431699', '33776923', '31434953', '29054837', '30015632', '33889306', '28522592', '34445380', '36702949', '26479923', '35075805', '37944531', '37000626', '34605222', '37143108', '28797031', '29472484', '31694835', '26926157', '30282693', '33199443', '35221336', '28473534', '35914528', '30596880', '29245952', '35972511', '34740372', '31488816', '26696773', '25437539', '36852691', '30629588', '37170307', '33602919', '37178682', '26846818', '37029129', '32641470', '32042320', '30992437', '30344100', '28581516', '27613526', '37116492', '35365682', '31693904', '36864254', '26095073', '33009951', '26270481', '30244973', '28232476', '34027491', '30380421', '28561063', '29458007', '26695443', '35802820', '31761724', '35410383', '25444907', '32884042', '32294323', '35508177', '30017245', '30146332', '34011980', '31883794', '30395907', '36058001', '33421710', '32075943', '29079660', '31036555', '29093017', '35383171', '30232224', '33852917', '32792481', '34250755', '32414908', '29625057', '26124487', '35180770', '30213835', '27381626', '27750381', '37452026', '36657446', '32183023', '30859564', '36223547']\n",
    "\n",
    "def get_pmids_from_cm():\n",
    "    url = \"https://www.cancermodels.org/api/model_metadata?select=model_id,data_source,provider_name,pubmed_ids\"\n",
    "    df = pd.read_json(url)\n",
    "    df = df[~df['pubmed_ids'].isna()]\n",
    "    df['pubmed_ids'] = df['pubmed_ids'].str.split(',')\n",
    "    df = df.explode('pubmed_ids')\n",
    "    df['pubmed_ids'] = df['pubmed_ids'].str.strip()\n",
    "    df = df.drop_duplicates('pubmed_ids')\n",
    "    df = df[df['pubmed_ids'] != \"\"]\n",
    "    df['pmid'] = [pmid.replace('PMID: ', '').replace('PMID:', '') for pmid in df['pubmed_ids']]\n",
    "    df = df.drop_duplicates('pmid')\n",
    "    return df\n",
    "\n",
    "df = get_pmids_from_cm()"
   ],
   "id": "f799fb578f5f1c93",
   "outputs": [],
   "execution_count": 3
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-12-11T11:21:21.646698Z",
     "start_time": "2024-12-11T11:21:21.625834Z"
    }
   },
   "cell_type": "code",
   "source": [
    "for pmid in pmids_positive:\n",
    "    if pmid not in df['pmid'].to_list():\n",
    "        print(f\"{pmid} is missing in CancerModels.Org\")"
   ],
   "id": "2c41f0e4a51cfc58",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32553164 is missing in CancerModels.Org\n",
      "33776923 is missing in CancerModels.Org\n",
      "34445380 is missing in CancerModels.Org\n",
      "36702949 is missing in CancerModels.Org\n",
      "35075805 is missing in CancerModels.Org\n",
      "37944531 is missing in CancerModels.Org\n",
      "37000626 is missing in CancerModels.Org\n",
      "34605222 is missing in CancerModels.Org\n",
      "37143108 is missing in CancerModels.Org\n",
      "29472484 is missing in CancerModels.Org\n",
      "33199443 is missing in CancerModels.Org\n",
      "30596880 is missing in CancerModels.Org\n",
      "35972511 is missing in CancerModels.Org\n",
      "34740372 is missing in CancerModels.Org\n",
      "31488816 is missing in CancerModels.Org\n",
      "37170307 is missing in CancerModels.Org\n",
      "33602919 is missing in CancerModels.Org\n",
      "37178682 is missing in CancerModels.Org\n",
      "32641470 is missing in CancerModels.Org\n",
      "36864254 is missing in CancerModels.Org\n",
      "35802820 is missing in CancerModels.Org\n",
      "31761724 is missing in CancerModels.Org\n",
      "32294323 is missing in CancerModels.Org\n",
      "35508177 is missing in CancerModels.Org\n",
      "31883794 is missing in CancerModels.Org\n",
      "36058001 is missing in CancerModels.Org\n",
      "31036555 is missing in CancerModels.Org\n",
      "30232224 is missing in CancerModels.Org\n",
      "33852917 is missing in CancerModels.Org\n",
      "32414908 is missing in CancerModels.Org\n",
      "35180770 is missing in CancerModels.Org\n",
      "30213835 is missing in CancerModels.Org\n",
      "36657446 is missing in CancerModels.Org\n"
     ]
    }
   ],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-12-11T11:20:32.517440Z",
     "start_time": "2024-12-11T11:20:32.509311Z"
    }
   },
   "cell_type": "code",
   "source": "",
   "id": "645c50b0e9395127",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'32990596'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 15
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
